diff -urBb dbus-1.12.8-orig/test/Makefile.in dbus-1.12.8/test/Makefile.in
--- dbus-1.12.8-orig/test/Makefile.in	2018-04-30 16:54:40.000000000 +0400
+++ dbus-1.12.8/test/Makefile.in	2018-05-23 10:14:17.939704200 +0400
@@ -2333,7 +2333,7 @@
 @DBUS_ENABLE_INSTALLED_TESTS_TRUE@		$(installable_tests); }
 
 all-local: copy-config-local uninstalled-config-local
-	$(AM_V_at)$(MKDIR_P) -m 700 XDG_RUNTIME_DIR
+	$(AM_V_at)$(MKDIR_P) XDG_RUNTIME_DIR
 
 copy-config-local:
 	$(AM_V_at)$(MKDIR_P) data/valid-config-files/session.d
