Только в mariadb-connector-c-2.1.0-src/include: my_config.h
diff -urBb mariadb-connector-c-2.1.0-src-orig/include/my_global.h mariadb-connector-c-2.1.0-src/include/my_global.h
--- mariadb-connector-c-2.1.0-src-orig/include/my_global.h	2015-01-29 21:21:48.000000000 +0300
+++ mariadb-connector-c-2.1.0-src/include/my_global.h	2015-07-14 17:54:18.643162800 +0300
@@ -454,7 +454,7 @@
 #ifdef _WIN32
 #define NO_DIR_LIBRARY		/* Not standar dir-library */
 #define USE_MY_STAT_STRUCT	/* For my_lib */
-#ifdef _SIZE_T_DEFINED
+#ifndef _SSIZE_T_DEFINED
 typedef SSIZE_T ssize_t;
 #endif
 #endif
diff -urBb mariadb-connector-c-2.1.0-src-orig/include/mysql_version.h mariadb-connector-c-2.1.0-src/include/mysql_version.h
--- mariadb-connector-c-2.1.0-src-orig/include/mysql_version.h	2015-01-29 21:22:09.000000000 +0300
+++ mariadb-connector-c-2.1.0-src/include/mysql_version.h	2015-07-14 17:48:25.696975400 +0300
@@ -17,8 +17,8 @@
 #define MYSQL_CONFIG_NAME		"my"
 
 #define MARIADB_PACKAGE_VERSION "2.1.0"
-#define MARIADB_SYSTEM_TYPE "Linux"
-#define MARIADB_MACHINE_TYPE "x86_64"
+#define MARIADB_SYSTEM_TYPE "Windows"
+#define MARIADB_MACHINE_TYPE "AMD64"
 
 /* mysqld compile time options */
 #ifndef MYSQL_CHARSET
