diff -urBb openal-soft-1.18.2-orig/CMakeLists.txt openal-soft-1.18.2/CMakeLists.txt
--- openal-soft-1.18.2-orig/CMakeLists.txt	2017-09-24 17:41:50.000000000 +0400
+++ openal-soft-1.18.2/CMakeLists.txt	2018-05-24 20:41:21.153755000 +0400
@@ -965,7 +965,8 @@
 OPTION(ALSOFT_REQUIRE_MMDEVAPI "Require MMDevApi backend" OFF)
 IF(HAVE_WINDOWS_H)
     # Check MMSystem backend
-    CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H -D_WIN32_WINNT=0x0502)
+    set(CMAKE_REQUIRED_DEFINITIONS -D_WIN32_WINNT=0x0502)
+    CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H)
     IF(HAVE_MMSYSTEM_H)
         CHECK_SHARED_FUNCTION_EXISTS(waveOutOpen "windows.h;mmsystem.h" winmm "" HAVE_LIBWINMM)
         IF(HAVE_LIBWINMM)
