diff -urBb poppler-0.26.2-orig/cmake/modules/FindCairo.cmake poppler-0.26.2/cmake/modules/FindCairo.cmake
--- poppler-0.26.2-orig/cmake/modules/FindCairo.cmake	2014-04-26 19:37:22 +0400
+++ poppler-0.26.2/cmake/modules/FindCairo.cmake	2014-08-09 11:00:11 +0400
@@ -18,7 +18,7 @@
 
 else(CAIRO_INCLUDE_DIRS AND CAIRO_LIBRARIES)
 
-if(NOT WIN32)
+if(NOT WIN32 OR MINGW)
   # use pkg-config to get the directories and then use these values
   # in the FIND_PATH() and FIND_LIBRARY() calls
   find_package(PkgConfig REQUIRED)
@@ -29,10 +29,10 @@
   if(_pc_cairo_FOUND)
     set(CAIRO_FOUND TRUE)
   endif(_pc_cairo_FOUND)
-else(NOT WIN32)
+else(NOT WIN32 OR MINGW)
   # assume so, for now
   set(CAIRO_FOUND TRUE)
-endif(NOT WIN32)
+endif(NOT WIN32 OR MINGW)
 
 if(CAIRO_FOUND)
   # set it back as false
diff -urBb poppler-0.26.2-orig/cmake/modules/FindFontconfig.cmake poppler-0.26.2/cmake/modules/FindFontconfig.cmake
--- poppler-0.26.2-orig/cmake/modules/FindFontconfig.cmake	2014-04-26 19:37:22 +0400
+++ poppler-0.26.2/cmake/modules/FindFontconfig.cmake	2014-08-09 11:01:01 +0400
@@ -19,14 +19,14 @@
 
 else (FONTCONFIG_LIBRARIES AND FONTCONFIG_INCLUDE_DIR)
 
-  if (NOT WIN32)
+  if (NOT WIN32 OR MINGW)
     # use pkg-config to get the directories and then use these values
     # in the FIND_PATH() and FIND_LIBRARY() calls
     find_package(PkgConfig)
     pkg_check_modules(PC_FONTCONFIG fontconfig)
 
     set(FONTCONFIG_DEFINITIONS ${PC_FONTCONFIG_CFLAGS_OTHER})
-  endif (NOT WIN32)
+  endif (NOT WIN32 OR MINGW)
 
   find_path(FONTCONFIG_INCLUDE_DIR fontconfig/fontconfig.h
     PATHS
diff -urBb poppler-0.26.2-orig/cmake/modules/FindGLIB.cmake poppler-0.26.2/cmake/modules/FindGLIB.cmake
--- poppler-0.26.2-orig/cmake/modules/FindGLIB.cmake	2014-04-26 19:37:22 +0400
+++ poppler-0.26.2/cmake/modules/FindGLIB.cmake	2014-08-09 11:02:02 +0400
@@ -13,11 +13,11 @@
 
 include(FindPackageHandleStandardArgs)
 
-if (NOT WIN32)
+if (NOT WIN32 OR MINGW)
   find_package(PkgConfig REQUIRED)
 
   pkg_check_modules(GLIB2 "glib-2.0>=${GLIB_REQUIRED}" "gobject-2.0>=${GLIB_REQUIRED}" "gio-2.0>=${GLIB_REQUIRED}")
 
   find_package_handle_standard_args(GLib DEFAULT_MSG GLIB2_LIBRARIES GLIB2_CFLAGS)
 
-endif(NOT WIN32)
+endif(NOT WIN32 OR MINGW)
diff -urBb poppler-0.26.2-orig/cmake/modules/FindGTK.cmake poppler-0.26.2/cmake/modules/FindGTK.cmake
--- poppler-0.26.2-orig/cmake/modules/FindGTK.cmake	2014-04-26 19:37:22 +0400
+++ poppler-0.26.2/cmake/modules/FindGTK.cmake	2014-08-09 11:02:33 +0400
@@ -12,11 +12,11 @@
 
 include(FindPackageHandleStandardArgs)
 
-if (NOT WIN32)
+if (NOT WIN32 OR MINGW)
   find_package(PkgConfig REQUIRED)
 
   pkg_check_modules(GTK3 "gtk+-3.0>=3.8" "gdk-pixbuf-2.0")
 
   find_package_handle_standard_args(GTK DEFAULT_MSG GTK3_LIBRARIES GTK3_CFLAGS)
 
-endif(NOT WIN32)
+endif(NOT WIN32 OR MINGW)
diff -urBb poppler-0.26.2-orig/cmake/modules/FindLIBOPENJPEG.cmake poppler-0.26.2/cmake/modules/FindLIBOPENJPEG.cmake
--- poppler-0.26.2-orig/cmake/modules/FindLIBOPENJPEG.cmake	2014-04-26 19:37:22 +0400
+++ poppler-0.26.2/cmake/modules/FindLIBOPENJPEG.cmake	2014-08-09 11:51:59 +0400
@@ -23,7 +23,7 @@
   set(LIBOPENJPEG_LIBRARIES)
   set(LIBOPENJPEG_INCLUDE_DIR)
 
-  find_path (LIBOPENJPEG_INCLUDE_DIR openjpeg.h PATH_SUFFIXES openjpeg)
+  find_path (LIBOPENJPEG_INCLUDE_DIR openjpeg.h PATH_SUFFIXES openjpeg openjpeg-1.5)
   find_library(LIBOPENJPEG_LIBRARIES openjpeg)
   if(LIBOPENJPEG_INCLUDE_DIR AND LIBOPENJPEG_LIBRARIES)
 
diff -urBb poppler-0.26.2-orig/cmake/modules/PopplerMacros.cmake poppler-0.26.2/cmake/modules/PopplerMacros.cmake
--- poppler-0.26.2-orig/cmake/modules/PopplerMacros.cmake	2014-04-26 19:37:22 +0400
+++ poppler-0.26.2/cmake/modules/PopplerMacros.cmake	2014-08-09 11:04:14 +0400
@@ -43,10 +43,10 @@
 endmacro(POPPLER_ADD_UNITTEST)
 
 macro(POPPLER_CREATE_INSTALL_PKGCONFIG generated_file install_location)
-  if(NOT WIN32)
+  if(NOT WIN32 OR MINGW)
     configure_file(${generated_file}.cmake ${CMAKE_CURRENT_BINARY_DIR}/${generated_file} @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${generated_file} DESTINATION ${install_location})
-  endif(NOT WIN32)
+  endif(NOT WIN32 OR MINGW)
 endmacro(POPPLER_CREATE_INSTALL_PKGCONFIG)
 
 macro(SHOW_END_MESSAGE what value)
