diff -urBb ppl-0.10.2-orig/m4/ac_check_gmp.m4 ppl-0.10.2/m4/ac_check_gmp.m4
--- ppl-0.10.2-orig/m4/ac_check_gmp.m4	2009-04-18 09:03:19 +0400
+++ ppl-0.10.2/m4/ac_check_gmp.m4	2010-05-03 10:59:10 +0400
@@ -84,7 +84,7 @@
       || GMP_LIMB_BITS != mp_bits_per_limb) {
     std::cerr
       << "GMP header (gmp.h) and library (ligmp.*) bits-per-limb mismatch:\n"
-      << "header gives " << __GMP_BITS_PER_MP_LIMB << ";\n"
+      //<< "header gives " << __GMP_BITS_PER_MP_LIMB << ";\n"
       << "library gives " << mp_bits_per_limb << ".\n"
       << "This probably means you are on a bi-arch system and\n"
       << "you are compiling with the wrong header or linking with\n"
