diff -urBb harfbuzz-5.3.1-orig/docs/Makefile.am harfbuzz-5.3.1/docs/Makefile.am
--- harfbuzz-5.3.1-orig/docs/Makefile.am	2022-10-20 00:07:54.000000000 +0400
+++ harfbuzz-5.3.1/docs/Makefile.am	2023-01-03 02:21:47.099323200 +0400
@@ -104,7 +104,7 @@
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in meson.build
+EXTRA_DIST = version.xml.in meson.build
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
