diff -urBbN harfbuzz-2.7.4-orig/docs/Makefile.am harfbuzz-2.7.4/docs/Makefile.am
--- harfbuzz-2.7.4-orig/docs/Makefile.am	2020-12-27 03:59:00.000000000 +0400
+++ harfbuzz-2.7.4/docs/Makefile.am	2021-01-05 14:39:12.622802500 +0400
@@ -107,7 +107,7 @@
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in meson.build
+EXTRA_DIST = version.xml.in meson.build
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
diff -urBbN harfbuzz-2.7.4-orig/gtk-doc.make harfbuzz-2.7.4/gtk-doc.make
--- harfbuzz-2.7.4-orig/gtk-doc.make	1970-01-01 04:00:00.000000000 +0400
+++ harfbuzz-2.7.4/gtk-doc.make	2021-01-05 14:49:54.310440200 +0400
@@ -0,0 +1 @@
+# empty
