diff -urBb harfbuzz-1.9.0-orig/src/gen-def.py harfbuzz-1.9.0/src/gen-def.py
--- harfbuzz-1.9.0-orig/src/gen-def.py	2018-08-26 09:24:01.000000000 +0400
+++ harfbuzz-1.9.0/src/gen-def.py	2018-10-19 11:28:39.791943000 +0400
@@ -13,7 +13,7 @@
 headers_content = []
 for h in header_paths:
 	if h.endswith (".h"):
-		with io.open (h, encoding='utf-8') as f: headers_content.append (f.read ())
+		with io.open (h, encoding='latin1') as f: headers_content.append (f.read ())
 
 result = """EXPORTS
 %s
