diff -urBb nasm-2.13.03-orig/include/nasmlib.h nasm-2.13.03/include/nasmlib.h
--- nasm-2.13.03-orig/include/nasmlib.h	2018-02-08 01:40:15.000000000 +0400
+++ nasm-2.13.03/include/nasmlib.h	2018-05-24 00:42:33.495775000 +0400
@@ -191,8 +191,8 @@
  * seg_init: Initialise the segment-number allocator.
  * seg_alloc: allocate a hitherto unused segment number.
  */
-void pure_func seg_init(void);
-int32_t pure_func seg_alloc(void);
+void seg_init(void);
+int32_t seg_alloc(void);
 
 /*
  * many output formats will be able to make use of this: a standard
