diff -urBb avrdude-6.1-orig/Makefile.am avrdude-6.1/Makefile.am
--- avrdude-6.1-orig/Makefile.am	2014-03-13 03:03:19.000000000 +0400
+++ avrdude-6.1/Makefile.am	2015-10-12 22:24:53.993480000 +0300
@@ -58,17 +58,17 @@
 
 avrdude_CPPFLAGS = -DCONFIG_DIR=\"$(sysconfdir)\"
 
-libavrdude_a_CPPFLAGS = -DCONFIG_DIR=\"$(sysconfdir)\"
+libavrdude_la_CPPFLAGS = -DCONFIG_DIR=\"$(sysconfdir)\"
 
 avrdude_CFLAGS   = @ENABLE_WARNINGS@
 
-libavrdude_a_CFLAGS   = @ENABLE_WARNINGS@
+libavrdude_la_CFLAGS   = @ENABLE_WARNINGS@
 
-avrdude_LDADD  = $(top_builddir)/$(noinst_LIBRARIES) @LIBUSB_1_0@ @LIBUSB@ @LIBFTDI1@ @LIBFTDI@ @LIBHID@ @LIBELF@ @LIBPTHREAD@ -lm
+avrdude_LDADD  = $(top_builddir)/$(noinst_LTLIBRARIES) @LIBUSB_1_0@ @LIBUSB@ @LIBFTDI1@ @LIBFTDI@ @LIBHID@ @LIBELF@ @LIBPTHREAD@ -lm
 
 bin_PROGRAMS = avrdude
 
-noinst_LIBRARIES = libavrdude.a
+noinst_LTLIBRARIES = libavrdude.la
 
 # automake thinks these generated files should be in the distribution,
 # but this might cause trouble for some users, so we rather don't want
@@ -85,7 +85,7 @@
 	$(distdir)/config_gram.c \
 	$(distdir)/config_gram.h
 
-libavrdude_a_SOURCES = \
+libavrdude_la_SOURCES = \
 	config_gram.y \
 	lexer.l \
 	arduino.h \
