diff -urBb mysql-connector-c-6.1.5-src-orig/extra/CMakeLists.txt mysql-connector-c-6.1.5-src/extra/CMakeLists.txt
--- mysql-connector-c-6.1.5-src-orig/extra/CMakeLists.txt	2014-06-11 06:10:44 +0300
+++ mysql-connector-c-6.1.5-src/extra/CMakeLists.txt	2015-01-13 15:16:42 +0300
@@ -22,7 +22,7 @@
 
 IF(NOT CMAKE_CROSSCOMPILING)
  ADD_EXECUTABLE(comp_err comp_err.c)
- TARGET_LINK_LIBRARIES(comp_err mysys mysys_ssl)
+ TARGET_LINK_LIBRARIES(comp_err mysys mysys_ssl ${CMAKE_THREAD_LIBS_INIT})
  SET_TARGET_PROPERTIES(comp_err PROPERTIES LINKER_LANGUAGE CXX)
 ENDIF()
 
@@ -58,16 +58,16 @@
 
 
 MYSQL_ADD_EXECUTABLE(my_print_defaults my_print_defaults.c)
-TARGET_LINK_LIBRARIES(my_print_defaults  mysys mysys_ssl)
+TARGET_LINK_LIBRARIES(my_print_defaults  mysys mysys_ssl ${CMAKE_THREAD_LIBS_INIT})
 SET_TARGET_PROPERTIES(my_print_defaults PROPERTIES LINKER_LANGUAGE CXX)
 
 MYSQL_ADD_EXECUTABLE(perror perror.c)
 ADD_DEPENDENCIES(perror GenError)
-TARGET_LINK_LIBRARIES(perror mysys mysys_ssl)
+TARGET_LINK_LIBRARIES(perror mysys mysys_ssl ${CMAKE_THREAD_LIBS_INIT})
 SET_TARGET_PROPERTIES(perror PROPERTIES LINKER_LANGUAGE CXX)
 
 MYSQL_ADD_EXECUTABLE(resolveip EXCLUDE_FROM_ALL resolveip.c)
-TARGET_LINK_LIBRARIES(resolveip mysys mysys_ssl)
+TARGET_LINK_LIBRARIES(resolveip mysys mysys_ssl ${CMAKE_THREAD_LIBS_INIT})
 SET_TARGET_PROPERTIES(resolveip PROPERTIES LINKER_LANGUAGE CXX)
 IF(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
   INCLUDE(CheckFunctionExists)
diff -urBb mysql-connector-c-6.1.5-src-orig/include/my_pthread.h mysql-connector-c-6.1.5-src/include/my_pthread.h
--- mysql-connector-c-6.1.5-src-orig/include/my_pthread.h	2014-06-11 06:10:43 +0300
+++ mysql-connector-c-6.1.5-src/include/my_pthread.h	2015-01-13 10:42:40 +0300
@@ -31,7 +31,7 @@
 #define EXTERNC
 #endif /* __cplusplus */ 
 
-#if defined(_WIN32)
+#if defined(_WIN32) && defined(_MSC_VER)
 typedef CRITICAL_SECTION pthread_mutex_t;
 typedef DWORD		 pthread_t;
 typedef struct thread_attr {
@@ -367,7 +367,7 @@
 
 	/* READ-WRITE thread locking */
 
-#ifndef _WIN32  /* read/write locks using pthread */
+#ifndef _MSC_VER  /* read/write locks using pthread */
 #define rw_lock_t pthread_rwlock_t
 #define my_rwlock_init(A,B) pthread_rwlock_init((A),(B))
 #define rw_rdlock(A) pthread_rwlock_rdlock(A)
diff -urBb mysql-connector-c-6.1.5-src-orig/libmysql/CMakeLists.txt mysql-connector-c-6.1.5-src/libmysql/CMakeLists.txt
--- mysql-connector-c-6.1.5-src-orig/libmysql/CMakeLists.txt	2014-06-11 06:10:43 +0300
+++ mysql-connector-c-6.1.5-src/libmysql/CMakeLists.txt	2015-01-13 15:14:39 +0300
@@ -255,6 +255,7 @@
   MERGE_LIBRARIES(libmysql SHARED ${LIBS}
     EXPORTS ${CLIENT_API_FUNCTIONS}
     COMPONENT SharedLibraries)
+  TARGET_LINK_LIBRARIES(libmysql ${CMAKE_THREAD_LIBS_INIT})
   IF(UNIX)
     # libtool compatability
     IF(CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR APPLE)
diff -urBb mysql-connector-c-6.1.5-src-orig/mysys/my_wincond.c mysql-connector-c-6.1.5-src/mysys/my_wincond.c
--- mysql-connector-c-6.1.5-src-orig/mysys/my_wincond.c	2014-06-11 06:10:43 +0300
+++ mysql-connector-c-6.1.5-src/mysys/my_wincond.c	2015-01-13 10:42:40 +0300
@@ -16,7 +16,7 @@
 /*****************************************************************************
 ** The following is a simple implementation of posix conditions
 *****************************************************************************/
-#if defined(_WIN32)
+#if defined(_WIN32) && defined(_MSC_VER)
 
 #include "mysys_priv.h"
 #include <m_string.h>
diff -urBb mysql-connector-c-6.1.5-src-orig/mysys/my_winthread.c mysql-connector-c-6.1.5-src/mysys/my_winthread.c
--- mysql-connector-c-6.1.5-src-orig/mysys/my_winthread.c	2014-06-11 06:10:43 +0300
+++ mysql-connector-c-6.1.5-src/mysys/my_winthread.c	2015-01-13 10:42:40 +0300
@@ -16,7 +16,7 @@
 /*****************************************************************************
 ** Simulation of posix threads calls for Windows
 *****************************************************************************/
-#if defined (_WIN32)
+#if defined (_WIN32) && defined(_MSC_VER)
 #include "mysys_priv.h"
 #include <process.h>
 #include <signal.h>
diff -urBb mysql-connector-c-6.1.5-src-orig/mysys/thr_rwlock.c mysql-connector-c-6.1.5-src/mysys/thr_rwlock.c
--- mysql-connector-c-6.1.5-src-orig/mysys/thr_rwlock.c	2014-06-11 06:10:43 +0300
+++ mysql-connector-c-6.1.5-src/mysys/thr_rwlock.c	2015-01-13 10:42:40 +0300
@@ -16,7 +16,7 @@
 /* Synchronization - readers / writer thread locks */
 
 #include "mysys_priv.h"
-#ifdef _WIN32
+#if defined(_WIN32) && defined(_MSC_VER)
 #include <errno.h>
 
 static BOOL have_srwlock= FALSE;
diff -urBb mysql-connector-c-6.1.5-src-orig/tests/CMakeLists.txt mysql-connector-c-6.1.5-src/tests/CMakeLists.txt
--- mysql-connector-c-6.1.5-src-orig/tests/CMakeLists.txt	2014-06-11 06:10:44 +0300
+++ mysql-connector-c-6.1.5-src/tests/CMakeLists.txt	2015-01-13 15:19:17 +0300
@@ -18,7 +18,7 @@
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/include)
 
 ADD_EXECUTABLE(mysql_client_test mysql_client_test.c)
-TARGET_LINK_LIBRARIES(mysql_client_test mysqlclient)
+TARGET_LINK_LIBRARIES(mysql_client_test mysqlclient ${CMAKE_THREAD_LIBS_INIT})
 SET_TARGET_PROPERTIES(mysql_client_test PROPERTIES LINKER_LANGUAGE CXX)
 
 #
