diff -urBb mysql-connector-c-6.1.6-src-orig/cmake/os/Windows.cmake mysql-connector-c-6.1.6-src/cmake/os/Windows.cmake
--- mysql-connector-c-6.1.6-src-orig/cmake/os/Windows.cmake	2015-02-26 00:09:49.000000000 +0300
+++ mysql-connector-c-6.1.6-src/cmake/os/Windows.cmake	2015-07-14 14:54:35.909425700 +0300
@@ -32,8 +32,10 @@
 # avoid running system checks by using pre-cached check results
 # system checks are expensive on VS since every tiny program is to be compiled in 
 # a VC solution.
-GET_FILENAME_COMPONENT(_SCRIPT_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
-INCLUDE(${_SCRIPT_DIR}/WindowsCache.cmake)
+IF(NOT MINGW)
+  GET_FILENAME_COMPONENT(_SCRIPT_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
+  INCLUDE(${_SCRIPT_DIR}/WindowsCache.cmake)
+ENDIF(NOT MINGW)
 
 # We require at least Visual Studio 2013 (aka 12.0) which has version nr 1800.
 IF(NOT FORCE_UNSUPPORTED_COMPILER AND MSVC_VERSION LESS 1800)
