diff -urBb postgresql-11.3-orig/src/common/relpath.c postgresql-11.3/src/common/relpath.c
--- postgresql-11.3-orig/src/common/relpath.c	2019-05-07 00:46:18.000000000 +0400
+++ postgresql-11.3/src/common/relpath.c	2019-05-14 17:02:28.130217300 +0400
@@ -18,7 +18,7 @@
 #include "postgres_fe.h"
 #endif
 
-#include "catalog/pg_tablespace_d.h"
+#include "catalog/pg_tablespace.h"
 #include "common/relpath.h"
 #include "storage/backendid.h"
 
