diff -urBb libgnugetopt-1.3-orig/getopt.h libgnugetopt-1.3/getopt.h
--- libgnugetopt-1.3-orig/getopt.h	2004-06-24 18:11:23 +0400
+++ libgnugetopt-1.3/getopt.h	2012-11-29 22:11:53 +0400
@@ -99,14 +99,7 @@
 #define optional_argument	2
 
 #if defined (__STDC__) && __STDC__
-#ifdef __GNU_LIBRARY__
-/* Many other libraries have conflicting prototypes for getopt, with
-   differences in the consts, in stdlib.h.  To avoid compilation
-   errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int argc, char *const *argv, const char *shortopts);
-#else /* not __GNU_LIBRARY__ */
-extern int getopt ();
-#endif /* __GNU_LIBRARY__ */
 extern int getopt_long (int argc, char *const *argv, const char *shortopts,
 		        const struct option *longopts, int *longind);
 extern int getopt_long_only (int argc, char *const *argv,
