diff -urBb libpng-1.6.34-orig/scripts/dfn.awk libpng-1.6.34/scripts/dfn.awk
--- libpng-1.6.34-orig/scripts/dfn.awk	2015-03-07 21:53:14.000000000 +0300
+++ libpng-1.6.34/scripts/dfn.awk	2018-05-23 22:11:40.733938700 +0400
@@ -106,6 +106,8 @@
    #	#define name "John Smith"
    #
    while (1) {
+      # Remove trailing CR
+      sub(/\r$/, "", line)
       # While there is an @" remove it and the next "@
       if (line ~ /@"/) {
          if (line ~ /@".*"@/) {
