diff -urBb qtdeclarative-everywhere-src-5.15.2-orig/src/plugins/scenegraph/scenegraph.pro qtdeclarative-everywhere-src-5.15.2/src/plugins/scenegraph/scenegraph.pro
--- qtdeclarative-everywhere-src-5.15.2-orig/src/plugins/scenegraph/scenegraph.pro	2020-10-27 12:02:12.000000000 +0400
+++ qtdeclarative-everywhere-src-5.15.2/src/plugins/scenegraph/scenegraph.pro	2021-01-07 13:27:01.476637200 +0400
@@ -1,5 +1,4 @@
 TEMPLATE = subdirs
 QT_FOR_CONFIG += quick
-qtConfig(d3d12): SUBDIRS += d3d12
 qtConfig(openvg): SUBDIRS += openvg
 
