diff -urBb babl-0.1.2-orig/tests/Makefile.am babl-0.1.2/tests/Makefile.am
--- babl-0.1.2-orig/tests/Makefile.am	2009-11-24 06:07:18 +0300
+++ babl-0.1.2/tests/Makefile.am	2010-02-25 20:14:58 +0300
@@ -12,7 +12,7 @@
 TESTS_ENVIRONMENT = BABL_PATH=$(top_builddir)/extensions/.libs
 
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/babl
-AM_LDFLAGS  = -pthread
+AM_LDFLAGS  = -lpthread
 
 LDADD = $(top_builddir)/babl/libbabl-@BABL_API_VERSION@.la \
 	$(MATH_LIB)
diff -urBb babl-0.1.2-orig/tests/Makefile.in babl-0.1.2/tests/Makefile.in
--- babl-0.1.2-orig/tests/Makefile.in	2010-01-14 20:55:50 +0300
+++ babl-0.1.2/tests/Makefile.in	2010-02-25 20:15:26 +0300
@@ -338,7 +338,7 @@
 top_srcdir = @top_srcdir@
 TESTS_ENVIRONMENT = BABL_PATH=$(top_builddir)/extensions/.libs
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/babl
-AM_LDFLAGS = -pthread
+AM_LDFLAGS = -lpthread
 LDADD = $(top_builddir)/babl/libbabl-@BABL_API_VERSION@.la \
 	$(MATH_LIB)
 
