diff -urBb gcc-12.2.0-orig/libatomic/Makefile.am gcc-12.2.0/libatomic/Makefile.am
--- gcc-12.2.0-orig/libatomic/Makefile.am	2022-08-19 12:09:54.604688300 +0400
+++ gcc-12.2.0/libatomic/Makefile.am	2022-12-25 14:45:09.987508500 +0400
@@ -23,7 +23,7 @@
 ##  <http://www.gnu.org/licenses/>.
 
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-12.2.0-orig/libatomic/Makefile.in gcc-12.2.0/libatomic/Makefile.in
--- gcc-12.2.0-orig/libatomic/Makefile.in	2022-08-19 12:09:54.604688300 +0400
+++ gcc-12.2.0/libatomic/Makefile.in	2022-12-25 14:45:09.989501300 +0400
@@ -385,7 +385,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) \
 	$(top_srcdir) $(top_builddir)
diff -urBb gcc-12.2.0-orig/libatomic/configure gcc-12.2.0/libatomic/configure
--- gcc-12.2.0-orig/libatomic/configure	2022-08-19 12:09:54.604688300 +0400
+++ gcc-12.2.0/libatomic/configure	2022-12-25 14:45:09.995456200 +0400
@@ -15751,9 +15751,7 @@
   multilib_arg=
 fi
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
-
-ac_config_files="$ac_config_files testsuite/libatomic-site-extra.exp"
+ac_config_files="$ac_config_files Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -16825,8 +16823,6 @@
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "gstdint.h") CONFIG_COMMANDS="$CONFIG_COMMANDS gstdint.h" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
-    "testsuite/libatomic-site-extra.exp") CONFIG_FILES="$CONFIG_FILES testsuite/libatomic-site-extra.exp" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-12.2.0-orig/libatomic/configure.ac gcc-12.2.0/libatomic/configure.ac
--- gcc-12.2.0-orig/libatomic/configure.ac	2022-08-19 12:09:54.604688300 +0400
+++ gcc-12.2.0/libatomic/configure.ac	2022-12-25 14:45:09.999473000 +0400
@@ -303,6 +303,5 @@
   multilib_arg=
 fi
 
-AC_CONFIG_FILES(Makefile testsuite/Makefile)
-AC_CONFIG_FILES(testsuite/libatomic-site-extra.exp)
+AC_CONFIG_FILES(Makefile)
 AC_OUTPUT
diff -urBb gcc-12.2.0-orig/libgomp/Makefile.am gcc-12.2.0/libgomp/Makefile.am
--- gcc-12.2.0-orig/libgomp/Makefile.am	2022-08-19 12:15:35.209048300 +0400
+++ gcc-12.2.0/libgomp/Makefile.am	2022-12-25 14:45:10.001440400 +0400
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = info-in-builddir
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-12.2.0-orig/libgomp/Makefile.in gcc-12.2.0/libgomp/Makefile.in
--- gcc-12.2.0-orig/libgomp/Makefile.in	2022-08-19 12:15:35.209048300 +0400
+++ gcc-12.2.0/libgomp/Makefile.in	2022-12-25 14:45:10.003434400 +0400
@@ -523,7 +523,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = info-in-builddir
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir) \
 	      $(top_srcdir)/../include
diff -urBb gcc-12.2.0-orig/libgomp/configure gcc-12.2.0/libgomp/configure
--- gcc-12.2.0-orig/libgomp/configure	2022-08-19 12:15:35.213048300 +0400
+++ gcc-12.2.0/libgomp/configure	2022-12-25 14:45:10.009446200 +0400
@@ -17066,11 +17066,9 @@
 
 ac_config_files="$ac_config_files omp.h omp_lib.h omp_lib.f90 libgomp_f.h"
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile libgomp.spec"
+ac_config_files="$ac_config_files Makefile libgomp.spec"
 
-ac_config_files="$ac_config_files testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in"
-
-ac_config_files="$ac_config_files testsuite/libgomp-site-extra.exp"
+ac_config_files="$ac_config_files"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -18218,10 +18216,7 @@
     "omp_lib.f90") CONFIG_FILES="$CONFIG_FILES omp_lib.f90" ;;
     "libgomp_f.h") CONFIG_FILES="$CONFIG_FILES libgomp_f.h" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
     "libgomp.spec") CONFIG_FILES="$CONFIG_FILES libgomp.spec" ;;
-    "testsuite/libgomp-test-support.pt.exp") CONFIG_FILES="$CONFIG_FILES testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in" ;;
-    "testsuite/libgomp-site-extra.exp") CONFIG_FILES="$CONFIG_FILES testsuite/libgomp-site-extra.exp" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-12.2.0-orig/libgomp/configure.ac gcc-12.2.0/libgomp/configure.ac
--- gcc-12.2.0-orig/libgomp/configure.ac	2022-08-19 12:09:55.128695000 +0400
+++ gcc-12.2.0/libgomp/configure.ac	2022-12-25 14:45:10.013408000 +0400
@@ -473,7 +473,5 @@
 GCC_BASE_VER
 
 AC_CONFIG_FILES(omp.h omp_lib.h omp_lib.f90 libgomp_f.h)
-AC_CONFIG_FILES(Makefile testsuite/Makefile libgomp.spec)
-AC_CONFIG_FILES([testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in])
-AC_CONFIG_FILES([testsuite/libgomp-site-extra.exp])
+AC_CONFIG_FILES(Makefile libgomp.spec)
 AC_OUTPUT
diff -urBb gcc-12.2.0-orig/libiberty/Makefile.in gcc-12.2.0/libiberty/Makefile.in
--- gcc-12.2.0-orig/libiberty/Makefile.in	2022-08-19 12:09:55.180695700 +0400
+++ gcc-12.2.0/libiberty/Makefile.in	2022-12-25 14:45:10.015402600 +0400
@@ -98,7 +98,7 @@
 	"tooldir=$(tooldir)"
 
 # Subdirectories to recurse into. We need to override this during cleaning
-SUBDIRS = testsuite
+SUBDIRS = 
 
 # FIXME: add @BUILD_INFO@ once we're sure it works for everyone.
 all: stamp-picdir stamp-noasandir $(TARGETLIB) required-list all-subdir
diff -urBb gcc-12.2.0-orig/libiberty/configure gcc-12.2.0/libiberty/configure
--- gcc-12.2.0-orig/libiberty/configure	2022-08-19 12:09:55.180695700 +0400
+++ gcc-12.2.0/libiberty/configure	2022-12-25 14:45:10.019392100 +0400
@@ -7723,7 +7723,7 @@
 
 
 # We need multilib support, but only if configuring for the target.
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
+ac_config_files="$ac_config_files Makefile"
 
 ac_config_commands="$ac_config_commands default"
 
@@ -8438,7 +8438,6 @@
   case $ac_config_target in
     "config.h") CONFIG_HEADERS="$CONFIG_HEADERS config.h:config.in" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
     "default") CONFIG_COMMANDS="$CONFIG_COMMANDS default" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
diff -urBb gcc-12.2.0-orig/libiberty/configure.ac gcc-12.2.0/libiberty/configure.ac
--- gcc-12.2.0-orig/libiberty/configure.ac	2022-08-19 12:09:55.180695700 +0400
+++ gcc-12.2.0/libiberty/configure.ac	2022-12-25 14:45:10.021415700 +0400
@@ -754,7 +754,7 @@
 AC_SUBST(htmldir)
 
 # We need multilib support, but only if configuring for the target.
-AC_CONFIG_FILES([Makefile testsuite/Makefile])
+AC_CONFIG_FILES([Makefile])
 AC_CONFIG_COMMANDS([default],
   [[test -z "$CONFIG_HEADERS" || echo timestamp > stamp-h
 if test -n "$CONFIG_FILES"; then
diff -urBb gcc-12.2.0-orig/libstdc++-v3/Makefile.am gcc-12.2.0/libstdc++-v3/Makefile.am
--- gcc-12.2.0-orig/libstdc++-v3/Makefile.am	2022-08-19 12:09:55.380698300 +0400
+++ gcc-12.2.0/libstdc++-v3/Makefile.am	2022-12-25 14:45:10.023386100 +0400
@@ -24,7 +24,7 @@
 
 if GLIBCXX_HOSTED
 ## Note that python must come after src.
-  hosted_source = src doc po testsuite python
+  hosted_source = src doc po python
 endif
 
 ## Keep this list sync'd with acinclude.m4:GLIBCXX_CONFIGURE.
@@ -32,10 +32,6 @@
 
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 
-# Testsuite/check forwarding targets.
-check-%:
-	cd testsuite && $(MAKE) $@
-
 # Documentation forwarding targets.
 doc-%:
 	cd doc && $(MAKE) $@
diff -urBb gcc-12.2.0-orig/libstdc++-v3/Makefile.in gcc-12.2.0/libstdc++-v3/Makefile.in
--- gcc-12.2.0-orig/libstdc++-v3/Makefile.in	2022-08-19 12:09:55.380698300 +0400
+++ gcc-12.2.0/libstdc++-v3/Makefile.in	2022-12-25 14:45:10.025376000 +0400
@@ -180,7 +180,7 @@
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = include libsupc++ src doc po testsuite python
+DIST_SUBDIRS = include libsupc++ src doc po python
 ABI_TWEAKS_SRCDIR = @ABI_TWEAKS_SRCDIR@
 ACLOCAL = @ACLOCAL@
 ALLOCATOR_H = @ALLOCATOR_H@
@@ -420,7 +420,7 @@
 
 # -I/-D flags to pass when compiling.
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES) $(CPPFLAGS)
-@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po testsuite python
+@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po python
 SUBDIRS = include libsupc++ $(hosted_source)
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 @BUILD_XML_FALSE@STAMP_XML = 
@@ -775,10 +775,6 @@
 .PRECIOUS: Makefile
 
 
-# Testsuite/check forwarding targets.
-check-%:
-	cd testsuite && $(MAKE) $@
-
 # Documentation forwarding targets.
 doc-%:
 	cd doc && $(MAKE) $@
diff -urBb gcc-12.2.0-orig/libstdc++-v3/configure gcc-12.2.0/libstdc++-v3/configure
--- gcc-12.2.0-orig/libstdc++-v3/configure	2022-08-19 12:09:55.416698700 +0400
+++ gcc-12.2.0/libstdc++-v3/configure	2022-12-25 14:46:59.037557700 +0400
@@ -5551,7 +5551,7 @@
   # expandable list at autoconf time; the second provides an expandable list
   # (i.e., shell variable) at configure time.
 
-  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/c++17 src/c++20 src/filesystem src/libbacktrace doc po testsuite python'
+  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/c++17 src/c++20 src/filesystem src/libbacktrace doc po python'
 
   # These need to be absolute paths, yet at the same time need to
   # canonicalize only relative paths, because then amd will not unmount
@@ -78903,7 +78903,7 @@
 # append it here.  Only modify Makefiles that have just been created.
 #
 # Also, get rid of this simulated-VPATH thing that automake does.
-ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/c++17/Makefile src/c++20/Makefile src/filesystem/Makefile src/libbacktrace/Makefile doc/Makefile po/Makefile testsuite/Makefile python/Makefile"
+ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/c++17/Makefile src/c++20/Makefile src/filesystem/Makefile src/libbacktrace/Makefile doc/Makefile po/Makefile python/Makefile"
 
 
 ac_config_commands="$ac_config_commands generate-headers"
@@ -80173,7 +80173,6 @@
     "src/libbacktrace/Makefile") CONFIG_FILES="$CONFIG_FILES src/libbacktrace/Makefile" ;;
     "doc/Makefile") CONFIG_FILES="$CONFIG_FILES doc/Makefile" ;;
     "po/Makefile") CONFIG_FILES="$CONFIG_FILES po/Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
     "python/Makefile") CONFIG_FILES="$CONFIG_FILES python/Makefile" ;;
     "generate-headers") CONFIG_COMMANDS="$CONFIG_COMMANDS generate-headers" ;;
 
diff -urBb gcc-12.2.0-orig/libstdc++-v3/configure.ac gcc-12.2.0/libstdc++-v3/configure.ac
--- gcc-12.2.0-orig/libstdc++-v3/configure.ac	2022-08-19 12:09:55.420698800 +0400
+++ gcc-12.2.0/libstdc++-v3/configure.ac	2022-12-25 14:45:10.057291200 +0400
@@ -478,9 +478,6 @@
 GLIBCXX_CHECK_GETENTROPY
 GLIBCXX_CHECK_ARC4RANDOM
 
-# This depends on GLIBCXX_ENABLE_SYMVERS and GLIBCXX_IS_NATIVE.
-GLIBCXX_CONFIGURE_TESTSUITE
-
 # For gthread support.  Depends on GLIBCXX_ENABLE_SYMVERS.
 GLIBCXX_CHECK_GTHREADS
 
diff -urBb gcc-12.2.0-orig/libvtv/Makefile.am gcc-12.2.0/libvtv/Makefile.am
--- gcc-12.2.0-orig/libvtv/Makefile.am	2022-08-19 12:09:55.756703100 +0400
+++ gcc-12.2.0/libvtv/Makefile.am	2022-12-25 14:45:10.059287100 +0400
@@ -19,12 +19,6 @@
 ## along with this library; see the file COPYING3.  If not see
 ## <http://www.gnu.org/licenses/>.
 
-if ENABLE_VTABLE_VERIFY
-  SUBDIRS = testsuite
-else
-  SUBDIRS=
-endif
-
 ACLOCAL_AMFLAGS = -I .. -I ../config
 
 # May be used by toolexeclibdir.
diff -urBb gcc-12.2.0-orig/libvtv/Makefile.in gcc-12.2.0/libvtv/Makefile.in
--- gcc-12.2.0-orig/libvtv/Makefile.in	2022-08-19 12:15:35.229048500 +0400
+++ gcc-12.2.0/libvtv/Makefile.in	2022-12-25 14:45:10.061280200 +0400
@@ -264,7 +264,6 @@
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = testsuite
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
@@ -405,8 +404,6 @@
 top_srcdir = @top_srcdir@
 toplevel_builddir = @toplevel_builddir@
 toplevel_srcdir = @toplevel_srcdir@
-@ENABLE_VTABLE_VERIFY_FALSE@SUBDIRS = 
-@ENABLE_VTABLE_VERIFY_TRUE@SUBDIRS = testsuite
 ACLOCAL_AMFLAGS = -I .. -I ../config
 
 # May be used by toolexeclibdir.
diff -urBb gcc-12.2.0-orig/libvtv/configure gcc-12.2.0/libvtv/configure
--- gcc-12.2.0-orig/libvtv/configure	2022-08-19 12:15:35.233048600 +0400
+++ gcc-12.2.0/libvtv/configure	2022-12-25 14:45:10.067264100 +0400
@@ -15748,11 +15748,6 @@
 ac_config_files="$ac_config_files Makefile"
 
 
-if test "x$VTV_SUPPORTED" = "xyes"; then
-  ac_config_files="$ac_config_files testsuite/Makefile"
-
-fi
-
 case "$target_os" in
   cygwin*|mingw32*)
     vtv_cygmin="yes"
@@ -16947,7 +16942,6 @@
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-12.2.0-orig/libvtv/configure.ac gcc-12.2.0/libvtv/configure.ac
--- gcc-12.2.0-orig/libvtv/configure.ac	2022-08-19 12:09:55.760703100 +0400
+++ gcc-12.2.0/libvtv/configure.ac	2022-12-25 14:45:10.073248300 +0400
@@ -174,21 +174,6 @@
 
 AC_CONFIG_FILES([Makefile])
 
-if test "x$VTV_SUPPORTED" = "xyes"; then
-  AC_CONFIG_FILES(AC_FOREACH([DIR], [testsuite], [DIR/Makefile ]), 
-    [cat > vpsed$$ << \_EOF
-s!`test -f '$<' || echo '$(srcdir)/'`!!
-_EOF
-    sed -f vpsed$$ $ac_file > tmp$$
-    mv tmp$$ $ac_file
-    rm vpsed$$
-    echo 'MULTISUBDIR =' >> $ac_file
-    ml_norecursion=yes
-    . ${multi_basedir}/config-ml.in
-    AS_UNSET([ml_norecursion])
-])
-fi
-
 case "$target_os" in
   cygwin*|mingw32*)
     vtv_cygmin="yes"
