diff -urBb gcc-12.2.0-orig/libiberty/Makefile.in gcc-12.2.0/libiberty/Makefile.in
--- gcc-12.2.0-orig/libiberty/Makefile.in	2022-12-25 14:50:01.607629200 +0400
+++ gcc-12.2.0/libiberty/Makefile.in	2022-12-25 14:50:41.317885800 +0400
@@ -376,7 +376,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
