diff -urBb gtkimageview-1.6.4-orig/configure.in gtkimageview-1.6.4/configure.in
--- gtkimageview-1.6.4-orig/configure.in	2009-04-06 00:02:43 +0400
+++ gtkimageview-1.6.4/configure.in	2014-08-08 08:49:37 +0400
@@ -39,7 +39,7 @@
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
-CFLAGS="${CFLAGS} -Wall -Werror -std=c99 -Wmissing-prototypes"
+CFLAGS="${CFLAGS} -Wall -std=c99 -Wmissing-prototypes"
 
 
 ######################################################################
