diff -urBb gtkimageview-1.6.4-orig/Makefile.am gtkimageview-1.6.4/Makefile.am
--- gtkimageview-1.6.4-orig/Makefile.am	2009-04-05 03:25:21 +0400
+++ gtkimageview-1.6.4/Makefile.am	2014-08-08 08:31:58 +0400
@@ -1,6 +1,6 @@
 # src must come before docs so that gtk-doc can build an executable
 # linked to the library.
-SUBDIRS = src tests docs
+SUBDIRS = src tests
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtkimageview.pc
diff -urBb gtkimageview-1.6.4-orig/configure.in gtkimageview-1.6.4/configure.in
--- gtkimageview-1.6.4-orig/configure.in	2009-04-06 00:02:43 +0400
+++ gtkimageview-1.6.4/configure.in	2014-08-08 08:51:12 +0400
@@ -18,8 +18,6 @@
 AM_PROG_LIBTOOL
 
 # Initialize GNOME environment
-GNOME_COMMON_INIT
-GNOME_COMPILE_WARNINGS(error)
 DEPRECATED_FLAGS="-DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
 AC_SUBST(DEPRECATED_FLAGS)
 
@@ -32,10 +30,6 @@
 AC_PATH_PROG(GLIB_GENMARSHAL, glib-genmarshal)
 AC_PATH_PROG(GLIB_MKENUMS, glib-mkenums)
 
-# 1.8 introduced the --rebuild-sections and --rebuild-types options to
-# 1.gtkdoc-scan that we need.
-GTK_DOC_CHECK([1.8])
-
 ######################################################################
 ##### Twiddle CFLAGS #################################################
 ######################################################################
@@ -48,8 +42,6 @@
 AC_CONFIG_FILES([
 gtkimageview.pc
 Makefile
-docs/Makefile
-docs/reference/Makefile
 src/Makefile
 tests/Makefile
 tests/testlib/Makefile
