diff -urBb lensfun-0.3.2-orig/CMakeLists.txt lensfun-0.3.2/CMakeLists.txt
--- lensfun-0.3.2-orig/CMakeLists.txt	2015-11-15 20:07:26.000000000 +0300
+++ lensfun-0.3.2/CMakeLists.txt	2016-01-08 13:12:04.304219100 +0300
@@ -80,7 +80,7 @@
     SET(GLIB2_BASE_DIR "${SOURCE_BASE_DIR}/glib-2.28.1" CACHE STRING "Base path of glib2 dir." FORCE)
   ENDIF()
   # install glib dll on windows next to lensfun.dll
-  IF (GLIB2_DLL)
+  IF (GLIB2_DLL AND MSVC)
     INSTALL(FILES ${GLIB2_DLL} DESTINATION ${CMAKE_INSTALL_BINDIR})
   ENDIF()
 ENDIF()
@@ -148,9 +148,6 @@
 
 # install include files
 INSTALL(FILES ${CMAKE_BINARY_DIR}/lensfun.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lensfun)
-IF(NOT HAVE_REGEX_H)
-  INSTALL(FILES include/regex/regex.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/regex)
-ENDIF()
 
 # install db files
 FILE(GLOB DATABASE_FILES data/db/*.xml)
@@ -167,7 +164,9 @@
             NAMES glib-2.dll glib-2-vs9.dll
             PATHS "${GLIB2_BASE_DIR}/bin"
             NO_SYSTEM_ENVIRONMENT_PATH)
+  IF(MSVC)
   INSTALL(FILES ${GLIB2_DLL} DESTINATION ${CMAKE_INSTALL_BINDIR})
+  ENDIF()
 ENDIF()
 
 # print some information
diff -urBb lensfun-0.3.2-orig/libs/regex/CMakeLists.txt lensfun-0.3.2/libs/regex/CMakeLists.txt
--- lensfun-0.3.2-orig/libs/regex/CMakeLists.txt	2015-11-15 20:07:26.000000000 +0300
+++ lensfun-0.3.2/libs/regex/CMakeLists.txt	2016-01-08 13:13:08.561260300 +0300
@@ -5,5 +5,4 @@
 ADD_LIBRARY(tre_regex STATIC ${REGEX_SRC})
 SET_TARGET_PROPERTIES(tre_regex PROPERTIES COMPILE_FLAGS "-DHAVE_SYS_TYPES_H -DTRE_REGEX_T_FIELD=value -DTRE_VERSION=\\\"0.8.0\\\" -Dinline=_inline")
 SET_TARGET_PROPERTIES(tre_regex PROPERTIES VERSION "${VERSION_MAJOR}.${VERSION_MINOR}).${VERSION_MICRO}")  
-INSTALL(TARGETS tre_regex ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 ENDIF()
