diff -urBb lensfun-0.2.8-orig/CMakeLists.txt lensfun-0.2.8/CMakeLists.txt
--- lensfun-0.2.8-orig/CMakeLists.txt	2013-06-23 14:37:01 +0400
+++ lensfun-0.2.8/CMakeLists.txt	2014-08-07 21:49:04 +0400
@@ -83,12 +83,12 @@
 ENDIF(BUILD_DOC)
 
 IF(WIN32)
+  IF(MSVC)
   # install into place in build-dir  
   SET(LENSFUN_INSTALL_PREFIX "${SOURCE_BASE_DIR}/lensfun" CACHE PATH "Install prefix for lensfun")    
-ELSE()
-  SET(LENSFUN_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" CACHE PATH "Install prefix for lensfun")    
-ENDIF()
-SET(CMAKE_INSTALL_PREFIX "${LENSFUN_INSTALL_PREFIX}" CACHE INTERNAL "" FORCE)
+    SET(CMAKE_INSTALL_PREFIX "${LENSFUN_INSTALL_PREFIX}" CACHE INTERNAL "" FORCE)
+  ENDIF(MSVC)
+ENDIF(WIN32)
 
 IF(WIN32)
   SET(PLATFORM_WINDOWS 1)  
@@ -135,9 +135,6 @@
 
 # install include files
 INSTALL(FILES ${CMAKE_BINARY_DIR}/lensfun.h DESTINATION ${INCLUDEDIR}/lensfun)
-IF(NOT HAVE_REGEX_H)
-  INSTALL(FILES include/regex/regex.h DESTINATION ${INCLUDEDIR}/regex)
-ENDIF()
 IF(BUILD_AUXFUN)
   INSTALL(FILES include/auxfun/image.h include/auxfun/rgbpixel.h DESTINATION ${INCLUDEDIR}/auxfun)
 ENDIF()
@@ -151,7 +148,7 @@
   ADD_SUBDIRECTORY(docs)
 ENDIF()
 
-IF(WIN32 AND NOT BUILD_STATIC)
+IF(WIN32 AND MSVC AND NOT BUILD_STATIC)
 FIND_FILE(GLIB2_DLL 
           NAMES glib-2.dll glib-2-vs9.dll
           PATHS "${GLIB2_BASE_DIR}/bin"
diff -urBb lensfun-0.2.8-orig/libs/regex/CMakeLists.txt lensfun-0.2.8/libs/regex/CMakeLists.txt
--- lensfun-0.2.8-orig/libs/regex/CMakeLists.txt	2013-03-10 14:50:45 +0400
+++ lensfun-0.2.8/libs/regex/CMakeLists.txt	2014-08-07 21:21:28 +0400
@@ -5,5 +5,4 @@
 ADD_LIBRARY(tre_regex STATIC ${REGEX_SRC})
 SET_TARGET_PROPERTIES(tre_regex PROPERTIES COMPILE_FLAGS "-DHAVE_SYS_TYPES_H -DTRE_REGEX_T_FIELD=value -DTRE_VERSION=\\\"0.8.0\\\" -Dinline=_inline")
 SET_TARGET_PROPERTIES(tre_regex PROPERTIES VERSION "${VERSION_MAJOR}.${VERSION_MINOR}).${VERSION_MICRO}")  
-INSTALL(TARGETS tre_regex ARCHIVE DESTINATION ${LIBDIR})
 ENDIF()
diff -urBb lensfun-0.2.8-orig/tests/CMakeLists.txt lensfun-0.2.8/tests/CMakeLists.txt
--- lensfun-0.2.8-orig/tests/CMakeLists.txt	2012-02-12 15:19:39 +0400
+++ lensfun-0.2.8/tests/CMakeLists.txt	2014-08-07 21:40:37 +0400
@@ -11,7 +11,9 @@
 ADD_EXECUTABLE(trwxml trwxml/trwxml.cpp)
 TARGET_LINK_LIBRARIES(trwxml lensfun ${COMMON_LIBS})
 
-INSTALL(TARGETS example tfun trwxml DESTINATION ${BINDIR})
+IF(MSVC)
+  INSTALL(TARGETS example tfun trwxml DESTINATION ${BINDIR})
+ENDIF(MSVC)
 
 IF(BUILD_AUXFUN)
 ADD_EXECUTABLE(tmod tmod/tmod.cpp)
