diff -urBb Python-2.7.17-orig/Makefile.pre.in Python-2.7.17/Makefile.pre.in
--- Python-2.7.17-orig/Makefile.pre.in	2020-05-26 09:32:32.244976900 +0400
+++ Python-2.7.17/Makefile.pre.in	2020-05-26 09:33:20.757991300 +0400
@@ -1187,11 +1187,7 @@
 	export EXE; EXE="$(BUILDEXE)"; \
 	if [ -n "$(MULTIARCH)" ]; then export MULTIARCH; MULTIARCH=$(MULTIARCH); fi; \
 	export PYTHON_FOR_BUILD; \
-	if [ "$(build)" = "$(host)" ]; then \
-	  PYTHON_FOR_BUILD="$(BUILDPYTHON)"; \
-	else \
 	  PYTHON_FOR_BUILD="$(PYTHON_FOR_BUILD)"; \
-	fi; \
 	cd $(srcdir)/Lib/$(PLATDIR); $(RUNSHARED) ./regen "$(CC)"
 
 python-config: $(srcdir)/Misc/python-config.in
