From https://gitlab.kitware.com/cmake/cmake/issues/16390

diff -urBb cmake-3.6.3-orig/Modules/FindSDL_sound.cmake cmake-3.6.3/Modules/FindSDL_sound.cmake
--- cmake-3.6.3-orig/Modules/FindSDL_sound.cmake	2016-11-03 17:45:22.000000000 +0400
+++ cmake-3.6.3/Modules/FindSDL_sound.cmake	2017-01-04 14:48:18.968354100 +0400
@@ -378,11 +378,10 @@
        endif()
      endif()
 
+     set(SDL_SOUND_LIBRARIES ${SDL_SOUND_EXTRAS} ${SDL_SOUND_LIBRARIES_TMP} CACHE INTERNAL "SDL_sound and dependent libraries")
    else()
-     set(SDL_SOUND_LIBRARIES "${SDL_SOUND_EXTRAS} ${SDL_SOUND_LIBRARY}" CACHE INTERNAL "SDL_sound and dependent libraries")
+     set(SDL_SOUND_LIBRARIES ${SDL_SOUND_EXTRAS} ${SDL_SOUND_LIBRARY} CACHE INTERNAL "SDL_sound and dependent libraries")
    endif()
-
-   set(SDL_SOUND_LIBRARIES "${SDL_SOUND_EXTRAS} ${SDL_SOUND_LIBRARIES_TMP}" CACHE INTERNAL "SDL_sound and dependent libraries")
  endif()
 
 if(SDL_SOUND_INCLUDE_DIR AND EXISTS "${SDL_SOUND_INCLUDE_DIR}/SDL_sound.h")
