diff -urBb cmake-2.8.8-orig/Modules/FindBZip2.cmake cmake-2.8.8/Modules/FindBZip2.cmake
--- cmake-2.8.8-orig/Modules/FindBZip2.cmake	2012-04-18 22:10:54 +0400
+++ cmake-2.8.8/Modules/FindBZip2.cmake	2012-06-29 21:52:25 +0400
@@ -45,8 +45,30 @@
                                   VERSION_VAR BZIP2_VERSION_STRING)
 
 IF (BZIP2_FOUND)
-   INCLUDE(CheckLibraryExists)
-   CHECK_LIBRARY_EXISTS("${BZIP2_LIBRARIES}" BZ2_bzCompressInit "" BZIP2_NEED_PREFIX)
+   
+   include(CheckCSourceCompiles)
+   
+   set(CMAKE_REQUIRED_LIBRARIES_save ${CMAKE_REQUIRED_LIBRARIES})
+   set(CMAKE_REQUIRED_INCLUDES_save ${CMAKE_REQUIRED_INCLUDES}) 
+   set(CMAKE_REQUIRED_LIBRARIES ${BZIP2_LIBRARIES})
+   set(CMAKE_REQUIRED_INCLUDES ${BZIP2_INCLUDE_DIR})
+   
+   CHECK_C_SOURCE_COMPILES("
+#include <bzlib.h>
+int main()
+{
+  return BZ2_bzlibVersion ();
+}" BZIP2_COMPILES)
+
+   if(BZIP2_COMPILES)
+      set(BZIP2_NEED_PREFIX TRUE)
+   endif(BZIP2_COMPILES)
+   
+   set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES_save})
+   set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES_save})
+   unset(CMAKE_REQUIRED_LIBRARIES_save)
+   unset(CMAKE_REQUIRED_INCLUDES_save)
+   unset(BZIP2_COMPILES)
 ENDIF (BZIP2_FOUND)
 
 MARK_AS_ADVANCED(BZIP2_INCLUDE_DIR)
