diff -urBb isl-0.11.1-orig/Makefile.am isl-0.11.1/Makefile.am
--- isl-0.11.1-orig/Makefile.am	2012-12-10 13:17:14 +0400
+++ isl-0.11.1/Makefile.am	2013-04-07 19:09:13 +0400
@@ -151,7 +151,7 @@
 	isl_sample_piplib.h \
 	isl_piplib.c
 libisl_la_LIBADD = @PIPLIB_LIBS@ @GMP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -no-undefined -version-info @versioninfo@ \
 	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@
 libisl_la_CPPFLAGS = $(INCLUDES) @PIPLIB_CPPFLAGS@ @GMP_CPPFLAGS@
 
