diff -urBb isl-0.13-orig/Makefile.am isl-0.13/Makefile.am
--- isl-0.13-orig/Makefile.am	2014-04-14 13:19:44 +0400
+++ isl-0.13/Makefile.am	2014-08-02 13:05:28 +0400
@@ -134,7 +134,7 @@
 	isl_vertices_private.h \
 	isl_vertices.c
 libisl_la_LIBADD = @GMP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -no-undefined -version-info @versioninfo@ \
 	@GMP_LDFLAGS@
 
 isl_test_LDFLAGS = @GMP_LDFLAGS@
