diff -urBb dmake-DMAKE_4_12_2_2-orig/configure dmake-DMAKE_4_12_2_2/configure
--- dmake-DMAKE_4_12_2_2-orig/configure	2014-08-30 03:41:10.000000000 +0400
+++ dmake-DMAKE_4_12_2_2/configure	2015-07-16 11:04:40.337919700 +0300
@@ -4349,7 +4349,7 @@
 	  OS_VERSION=cygwin
 	  CASE_INSENSITIVE_FS=true
 	  ;;
-	"MINGW")
+	"MINGW"*)
 	  OS_VERSION=mingw
 	  OS_TYPE=winnt
 	  BUILDINFO="Windows / MinGW"
diff -urBb dmake-DMAKE_4_12_2_2-orig/sysintf.c dmake-DMAKE_4_12_2_2/sysintf.c
--- dmake-DMAKE_4_12_2_2-orig/sysintf.c	2014-08-30 03:41:10.000000000 +0400
+++ dmake-DMAKE_4_12_2_2/sysintf.c	2015-07-16 12:26:21.614256700 +0300
@@ -649,7 +649,7 @@
 {
    int fd; /* file descriptor */
 
-#if defined(HAVE_MKSTEMP)
+#if defined(HAVE_MKSTEMP) && !defined(__MINGW32__)
    mode_t       mask;
 
    *path = DmStrJoin( tmpdir, DirSepStr, -1, FALSE);
@@ -704,6 +704,8 @@
 
    tmpdir = Read_env_string( "TMPDIR" );
    if( tmpdir == NIL(char) )
+      tmpdir = Read_env_string( "TEMP" );
+   if( tmpdir == NIL(char) )
       tmpdir = "/tmp";
 
    while( --tries )
