diff -urBb cairo-1.10.2-orig/configure.ac cairo-1.10.2/configure.ac
--- cairo-1.10.2-orig/configure.ac	2010-12-13 19:30:14 +0300
+++ cairo-1.10.2/configure.ac	2011-01-06 21:26:46 +0300
@@ -306,7 +306,12 @@
 	  AC_CHECK_HEADER(GL/gl.h,, [use_gl="no (gl.pc nor OpenGL headers not found)"])
 	  if test "x$use_gl" = "xyes"; then
 	      gl_NONPKGCONFIG_CFLAGS=
+		  if test "x$use_win32" != "xyes"; then
 	      gl_NONPKGCONFIG_LIBS="-lGL"
+		  else
+			gl_NONPKGCONFIG_LIBS="-lopengl32"
+			gl_REQUIRES=""
+		  fi
 	  fi])
 
   # Disable system GLEW on UNIX since it assumes X and crashes with EGL
