diff -urBb cairo-1.12.16-orig/build/configure.ac.warnings cairo-1.12.16/build/configure.ac.warnings
--- cairo-1.12.16-orig/build/configure.ac.warnings	2013-08-26 19:07:21 +0400
+++ cairo-1.12.16/build/configure.ac.warnings	2014-08-06 15:56:17 +0400
@@ -39,7 +39,7 @@
 dnl -flto working really needs a test link, not just a compile
 
 safe_MAYBE_WARN="$MAYBE_WARN"
-MAYBE_WARN="$MAYBE_WARN -flto"
+MAYBE_WARN="$MAYBE_WARN -flto -ffat-lto-objects -fuse-linker-plugin"
 AC_TRY_LINK([],[
 	int main(int argc, char **argv) { return 0; }
 ],[],[
