diff -urBb qtbase-everywhere-src-5.15.2-kde/src/corelib/io/qfilesystemengine_win.cpp qtbase-everywhere-src-5.15.2/src/corelib/io/qfilesystemengine_win.cpp
--- qtbase-everywhere-src-5.15.2-kde/src/corelib/io/qfilesystemengine_win.cpp	2020-10-27 12:02:11.000000000 +0400
+++ qtbase-everywhere-src-5.15.2/src/corelib/io/qfilesystemengine_win.cpp	2021-12-31 11:52:26.181694600 +0400
@@ -664,7 +664,7 @@
     return QFileSystemEntry(ret, QFileSystemEntry::FromInternalPath());
 }
 
-#if defined(Q_CC_MINGW) && WINVER < 0x0602 //  Windows 8 onwards
+#if defined(Q_CC_MINGW) && (WINVER < 0x0602 && _WIN32_WINNT < _WIN32_WINNT_WIN8) //  Windows 8 onwards
 
 typedef struct _FILE_ID_INFO {
     ULONGLONG VolumeSerialNumber;
