From https://bugreports.qt.io/browse/QTBUG-79033
Fixes issue with PostgreSQL 12: QSqlTableModel::select() return false.

--- a/src/plugins/sqldrivers/psql/qsql_psql.cpp
+++ b/src/plugins/sqldrivers/psql/qsql_psql.cpp
@@ -1440,7 +1440,7 @@
 
     QString stmt = QLatin1String("select pg_attribute.attname, pg_attribute.atttypid::int, "
                                  "pg_attribute.attnotnull, pg_attribute.attlen, pg_attribute.atttypmod, "
-                                 "pg_attrdef.adsrc "
+                                 "pg_get_expr(pg_attrdef.adbin, pg_attrdef.adrelid) "
                                  "from pg_class, pg_attribute "
                                  "left join pg_attrdef on (pg_attrdef.adrelid = "
                                  "pg_attribute.attrelid and pg_attrdef.adnum = pg_attribute.attnum) "
