diff -urBb qttools-everywhere-src-5.15.14-orig/src/assistant/qcollectiongenerator/main.c qttools-everywhere-src-5.15.14/src/assistant/qcollectiongenerator/main.c
--- qttools-everywhere-src-5.15.14-orig/src/assistant/qcollectiongenerator/main.c	2024-05-08 09:15:02.000000000 +0400
+++ qttools-everywhere-src-5.15.14/src/assistant/qcollectiongenerator/main.c	2024-07-06 09:34:42.264587300 +0400
@@ -98,7 +98,7 @@
 
     argv[0] = newPath;
 #ifdef _WIN32
-    const intptr_t ret = _spawnvp(_P_WAIT, newPath, argv);
+    const intptr_t ret = _spawnvp(_P_WAIT, newPath, (const char* const*)argv);
     if (ret == -1) {
         fprintf(stderr, "Error while executing \"%s\" tool.\n", newPath);
         return 3;
