diff -urBb libjxl-0.11.2-orig/.github/workflows/conformance.yml libjxl-0.11.2/.github/workflows/conformance.yml
--- libjxl-0.11.2-orig/.github/workflows/conformance.yml	2026-02-10 17:53:22.000000000 +0400
+++ libjxl-0.11.2/.github/workflows/conformance.yml	2026-05-31 02:58:55.857825600 +0400
@@ -63,6 +63,7 @@
         include:
           - name: AVX3
             cflags: -DHWY_DISABLED_TARGETS=HWY_AVX3-1
+            cmake_args: -DJPEGXL_ENABLE_HWY_AVX3=ON
           - name: AVX2
             cflags: -DHWY_DISABLED_TARGETS=HWY_AVX2-1
           - name: SSE4
Only in libjxl-0.11.2: 03e432d38f3e735028823379f234d54ee9ec73d9.patch
diff -urBb libjxl-0.11.2-orig/CMakeLists.txt libjxl-0.11.2/CMakeLists.txt
--- libjxl-0.11.2-orig/CMakeLists.txt	2026-02-10 17:53:22.000000000 +0400
+++ libjxl-0.11.2/CMakeLists.txt	2026-05-31 03:02:38.057973500 +0400
@@ -102,6 +102,12 @@
   set(ENABLE_SKCMS_DEFAULT YES)
 endif()
 
+set(JPEGXL_HWY_TARGETS AVX2 AVX3 AVX3_DL AVX3_SPR AVX3_ZEN4 EMU128 NEON NEON_BF16 NEON_WITHOUT_AES PPC10 PPC8 PPC9 RVV SCALAR SSE2 SSE4 SSSE3 SVE SVE_256 SVE2 SVE2_128 WASM WASM_EMU256 Z14 Z15)
+set(JPEGXL_HWY_TARGETS_OFF_BY_DEFAULT AVX3 AVX3_SPR AVX3_ZEN4 RVV SSSE3 SVE_256)
+if (NOT CXX_SVE_SUPPORTED)
+  list(APPEND JPEGXL_HWY_TARGETS_OFF_BY_DEFAULT SVE SVE2 SVE_256 SVE2_128)
+endif()
+
 # Standard cmake naming for building shared libraries.
 get_property(SHARED_LIBS_SUPPORTED GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS)
 option(BUILD_SHARED_LIBS "Build shared libraries instead of static ones" ${SHARED_LIBS_SUPPORTED})
@@ -148,8 +154,6 @@
     "Build third-party plugins to support JPEG XL in other applications.")
 set(JPEGXL_ENABLE_COVERAGE false CACHE BOOL
     "Enable code coverage tracking for libjxl. This also enables debug and disables optimizations.")
-set(JPEGXL_ENABLE_SIZELESS_VECTORS false CACHE BOOL
-    "Builds in support for SVE/RVV vectorization")
 set(JPEGXL_ENABLE_TRANSCODE_JPEG true CACHE BOOL
     "Builds in support for decoding transcoded JXL files back to JPEG,\
  disabling it makes the decoder reject JXL_DEC_JPEG_RECONSTRUCTION events,\
@@ -168,15 +172,19 @@
     "Set flags to enable NEON in arm if not enabled by your toolchain.")
 set(JPEGXL_TEST_TOOLS false CACHE BOOL
     "Run scripts that test the encoding / decoding tools.")
-set(JPEGXL_ENABLE_AVX512 false CACHE BOOL
-    "Build with AVX512 support (faster on CPUs that support it, but larger binary size).")
-set(JPEGXL_ENABLE_AVX512_SPR false CACHE BOOL
-    "Build with AVX-512FP16 support (faster on CPUs that support it, but larger binary size).")
-set(JPEGXL_ENABLE_AVX512_ZEN4 false CACHE BOOL
-"Build with Zen4-optimized AVX512 support (faster on CPUs that support it, but larger binary size).")
 set(JPEGXL_ENABLE_WASM_THREADS true CACHE BOOL
     "Builds WASM modules with threads support")
 
+# HWY codegen levers
+foreach(tgt IN LISTS JPEGXL_HWY_TARGETS)
+  set(JPEGXL_ENABLE_HWY_${tgt}_DEFAULT true)
+  if (${tgt} IN_LIST JPEGXL_HWY_TARGETS_OFF_BY_DEFAULT)
+    set(JPEGXL_ENABLE_HWY_${tgt}_DEFAULT false)
+  endif()
+  set(JPEGXL_ENABLE_HWY_${tgt} ${JPEGXL_ENABLE_HWY_${tgt}_DEFAULT} CACHE BOOL
+    "Build with ${tgt} support (faster on CPUs that support it, but larger binary size).")
+endforeach()
+
 # Force system dependencies.
 set(JPEGXL_FORCE_SYSTEM_BROTLI false CACHE BOOL
     "Force using system installed brotli instead of third_party/brotli source.")
@@ -198,23 +206,25 @@
     "Compiled IDs C:${CMAKE_C_COMPILER_ID}, C++:${CMAKE_CXX_COMPILER_ID}")
 
 set(JXL_HWY_INCLUDE_DIRS "$<BUILD_INTERFACE:$<TARGET_PROPERTY:$<IF:$<TARGET_EXISTS:hwy::hwy>,hwy::hwy,hwy>,INTERFACE_INCLUDE_DIRECTORIES>>")
-# Always disable SSSE3 since it is rare to have SSSE3 but not SSE4
-set(HWY_DISABLED_TARGETS "HWY_SSSE3")
-if (NOT JPEGXL_ENABLE_AVX512)
-  message(STATUS "Disabled AVX512 (set JPEGXL_ENABLE_AVX512 to enable it)")
-  set(HWY_DISABLED_TARGETS "${HWY_DISABLED_TARGETS}|HWY_AVX3")
+
+if (JXL_HWY_DISABLED_TARGETS_FORCED)
+  message(WARNING "HWY_DISABLED_TARGETS is overridden; JPEGXL_ENABLE_hwy_target will be ignored")
+else()
+  set(HWY_DISABLED_TARGETS_LIST "")
+  foreach(tgt IN LISTS JPEGXL_HWY_TARGETS)
+    message(STATUS "HWY ${tgt} enabled: ${JPEGXL_ENABLE_HWY_${tgt}}")
+    if (NOT JPEGXL_ENABLE_HWY_${tgt})
+      list(APPEND HWY_DISABLED_TARGETS_LIST HWY_${tgt})
+    endif()
+  endforeach()
+  if (NOT JPEGXL_ENABLE_HWY_AVX3)
   add_definitions(-DFJXL_ENABLE_AVX512=0)
+  endif()
+  if (HWY_DISABLED_TARGETS_LIST)
+    list(JOIN HWY_DISABLED_TARGETS_LIST "|" HWY_DISABLED_TARGETS)
+    add_compile_options($<$<NOT:$<COMPILE_LANGUAGE:RC>>:-DHWY_DISABLED_TARGETS=\(${HWY_DISABLED_TARGETS}\)>)
+  endif()
 endif()
-if (NOT JPEGXL_ENABLE_AVX512_SPR)
-  message(STATUS "Disabled AVX512_SPR (set JPEGXL_ENABLE_AVX512_SPR to enable it)")
-  set(HWY_DISABLED_TARGETS "${HWY_DISABLED_TARGETS}|HWY_AVX3_SPR")
-endif()
-if (NOT JPEGXL_ENABLE_AVX512_ZEN4)
-  message(STATUS "Disabled AVX512_ZEN4 (set JPEGXL_ENABLE_AVX512_ZEN4 to enable it)")
-  set(HWY_DISABLED_TARGETS "${HWY_DISABLED_TARGETS}|HWY_AVX3_ZEN4")
-endif()
-
-
 
 # CMAKE_EXPORT_COMPILE_COMMANDS is used to generate the compilation database
 # used by clang-tidy.
@@ -325,14 +335,6 @@
     -D__TIME__="redacted"
   )
 
-  # TODO(eustas): JXL currently compiles, but does not pass tests...
-  if (NOT JXL_HWY_DISABLED_TARGETS_FORCED)
-    if (NOT JPEGXL_ENABLE_SIZELESS_VECTORS)
-      set(HWY_DISABLED_TARGETS "${HWY_DISABLED_TARGETS}|HWY_SVE|HWY_SVE2|HWY_SVE_256|HWY_SVE2_128|HWY_RVV")
-    endif()
-    add_compile_options($<$<NOT:$<COMPILE_LANGUAGE:RC>>:-DHWY_DISABLED_TARGETS=\(${HWY_DISABLED_TARGETS}\)>)
-  endif()
-
   # Machine flags.
   add_compile_options(-funwind-tables)
   if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
