diff -urBb gcc-4.8.2-orig/libstdc++-v3/src/c++98/math_stubs_float.cc gcc-4.8.2/libstdc++-v3/src/c++98/math_stubs_float.cc
--- gcc-4.8.2-orig/libstdc++-v3/src/c++98/math_stubs_float.cc	2013-02-03 21:54:05 +0400
+++ gcc-4.8.2/libstdc++-v3/src/c++98/math_stubs_float.cc	2014-04-27 21:13:03 +0400
@@ -30,8 +30,7 @@
 
 extern "C" 
 {
-#ifndef _GLIBCXX_HAVE_FABSF
-  float
+#if !defined(_GLIBCXX_HAVE_FABSF) && !defined(_MATH_H_)
   fabsf(float x)
   {
     return (float) fabs(x);
@@ -94,7 +93,7 @@
   }
 #endif
 
-#ifndef _GLIBCXX_HAVE_EXPF
+#if !defined(_GLIBCXX_HAVE_EXPF) && !defined(_MATH_H_)
   float
   expf(float x)
   {
diff -urBb gcc-4.8.2-orig/libstdc++-v3/src/c++98/math_stubs_long_double.cc gcc-4.8.2/libstdc++-v3/src/c++98/math_stubs_long_double.cc
--- gcc-4.8.2-orig/libstdc++-v3/src/c++98/math_stubs_long_double.cc	2013-02-03 21:54:05 +0400
+++ gcc-4.8.2/libstdc++-v3/src/c++98/math_stubs_long_double.cc	2014-04-27 21:13:48 +0400
@@ -30,7 +30,7 @@
 
 extern "C" 
 {
-#ifndef _GLIBCXX_HAVE_FABSL
+#if !defined(_GLIBCXX_HAVE_FABSL) && !defined(_MATH_H_)
   long double
   fabsl(long double x)
   {
