diff -urBbN rhash-1.3.4-orig/librhash/Makefile rhash-1.3.4/librhash/Makefile
--- rhash-1.3.4-orig/librhash/Makefile	2016-11-06 19:49:40.000000000 +0400
+++ rhash-1.3.4/librhash/Makefile	2017-04-15 09:56:17.664805500 +0400
@@ -26,6 +26,7 @@
 DESTDIR =
 PREFIX  ?= /usr/local
 INCDIR  = $(PREFIX)/include
+BINDIR  = $(PREFIX)/bin
 LIBDIR  = $(PREFIX)/lib
 LIBRARY = librhash.a
 SONAME  = librhash.so.0
@@ -176,15 +177,22 @@
 
 # MINGW dll target
 DLLNAME = librhash.dll
+IMPLIBNAME = librhash.dll.a
 
 dll: $(DLLNAME)
 $(DLLNAME): $(SOURCES)
 	sed -n '1s/.*/{ global:/p; s/^RHASH_API.* \([a-z0-9_]\+\)(.*/  \1;/p; $$s/.*/local: *; };/p' $(LIB_HEADERS) > exports.sym
-	$(CC) -DRHASH_EXPORTS $(ALLCFLAGS) -shared $(SOURCES) -Wl,--version-script,exports.sym,--output-def,librhash.def,-soname,$@ $(LIBLDFLAGS) -o $@
+	$(CC) -DRHASH_EXPORTS $(ALLCFLAGS) -shared $(SOURCES) -Wl,--version-script,exports.sym,--output-def,librhash.def,-soname,$@,--out-implib,$(IMPLIBNAME) $(LIBLDFLAGS) -o $@
 
 test-dll: $(DLLNAME) test_hashes.o
 	$(CC) $(PRGCFLAGS) test_hashes.o $(DLLNAME) -o $(TEST_SHARED) && ./$(TEST_SHARED)
 
+install-dll: $(DLLNAME)
+	$(INSTALL) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)
+	$(INSTALL) $(DLLNAME) $(DESTDIR)$(BINDIR)/
+	$(INSTALL) $(IMPLIBNAME) $(DESTDIR)$(LIBDIR)/
+
 # shared and static libraries
 $(SONAME): $(SOURCES)
 	sed -n '1s/.*/{ global:/p; s/^RHASH_API.* \([a-z0-9_]\+\)(.*/  \1;/p; $$s/.*/local: *; };/p' $(SO_HEADERS) > exports.sym
@@ -211,4 +219,4 @@
 #	if [ -f $(DLLNAME) ]; then make test-dll; fi
 
 clean:
-	rm -f *.o $(LIBRARY) $(TEST_TARGET) $(TEST_SHARED) $(SONAME) $(SOLINK) exports.sym $(DLLNAME) librhash.def
+	rm -f *.o $(LIBRARY) $(TEST_TARGET) $(TEST_SHARED) $(SONAME) $(SOLINK) exports.sym $(DLLNAME) $(IMPLIBNAME) librhash.def
diff -urBbN rhash-1.3.4-orig/Makefile rhash-1.3.4/Makefile
--- rhash-1.3.4-orig/Makefile	2016-11-06 03:22:07.000000000 +0400
+++ rhash-1.3.4/Makefile	2017-04-15 09:43:29.179941900 +0400
@@ -54,6 +54,7 @@
 I18N_FILES = po/de.po po/en_AU.po po/es.po po/gl.po po/it.po po/ru.po
 DIST_FILES     = $(LIN_DIST_FILES) $(LIBRHASH_FILES) $(WIN_DIST_FILES) $(WIN_SRC_FILES) $(I18N_FILES)
 DESTDIR = 
+EXE = 
 BINDIR  = $(PREFIX)/bin
 MANDIR  = $(PREFIX)/share/man
 LOCALEDIR = $(PREFIX)/share/locale
@@ -111,7 +112,7 @@
 
 # dependencies should be properly set, otherwise 'make -j<n>' can fail
 install-symlinks: mkdir-bin install-man
-	for f in $(SYMLINKS); do ln -fs rhash $(DESTDIR)$(BINDIR)/$$f; done
+	for f in $(SYMLINKS); do ln -fs rhash $(DESTDIR)$(BINDIR)/$$f$(EXE); done
 	cd $(DESTDIR)$(MANDIR)/man1 && for f in $(SYMLINKS); do ln -fs rhash.1* $$f.1; done
 
 uninstall-binary:
