diff -urBb ufraw-0.16-orig/dcraw.cc ufraw-0.16/dcraw.cc
--- ufraw-0.16-orig/dcraw.cc	2009-10-13 05:47:11 +0400
+++ ufraw-0.16/dcraw.cc	2009-10-17 10:07:21 +0400
@@ -39,6 +39,7 @@
 #include <sys/types.h>
 #ifdef HAVE_LIBJPEG
 extern "C" {
+#define XMD_H	// say jpeglib about already defined INT32
 #include <jpeglib.h>
 }
 #endif
@@ -70,6 +71,8 @@
 #endif
 #ifdef WIN32
 #include <sys/utime.h>
+// don't include in windows.h rpcndr.h where declare boolean type
+#define _OBJC_NO_COM
 #include <winsock2.h>
 #ifndef __MINGW32__ /* causes warnings in cygwin's gcc -mno-cygwin UF*/
 #pragma comment(lib, "ws2_32.lib")
