diff -urBb ufraw-0.18/dcraw.cc tmp/ufraw-build/ufraw-0.18//dcraw.cc
--- ufraw-0.18/dcraw.cc	2011-02-20 11:56:15 +0400
+++ tmp/ufraw-build/ufraw-0.18//dcraw.cc	2013-01-11 21:20:58 +0400
@@ -40,11 +40,6 @@
 #include <string.h>
 #include <time.h>
 #include <sys/types.h>
-#ifdef HAVE_LIBJPEG
-extern "C" {
-#include <jpeglib.h>
-}
-#endif
 /*
    NO_LCMS disables the "-p" option.
  */
@@ -91,6 +86,12 @@
 typedef unsigned long long UINT64;
 #endif
 
+#ifdef HAVE_LIBJPEG
+extern "C" {
+#include <jpeglib.h>
+}
+#endif
+
 #ifdef LJPEG_DECODE
 #error Please compile dcraw.c by itself.
 #error Do not link it with ljpeg_decode.
