diff -urBb ca-certificates-orig/mozilla/certdata2pem.py ca-certificates/mozilla/certdata2pem.py
--- ca-certificates-orig/mozilla/certdata2pem.py	2022-12-06 13:23:25.000000000 +0400
+++ ca-certificates/mozilla/certdata2pem.py	2024-06-23 19:36:54.758068400 +0400
@@ -28,8 +28,6 @@
 import textwrap
 import io
 
-from cryptography import x509
-
 
 objects = []
 
@@ -122,12 +120,6 @@
         if not obj['CKA_LABEL'] in trust or not trust[obj['CKA_LABEL']]:
             continue
 
-        cert = x509.load_der_x509_certificate(bytes(obj['CKA_VALUE']))
-        if cert.not_valid_after < datetime.datetime.utcnow():
-            print('!'*74)
-            print('Trusted but expired certificate found: %s' % obj['CKA_LABEL'])
-            print('!'*74)
-
         bname = obj['CKA_LABEL'][1:-1].replace('/', '_')\
                                       .replace(' ', '_')\
                                       .replace('(', '=')\
