diff -urBb libwebp-0.5.1-orig/configure.ac libwebp-0.5.1/configure.ac
--- libwebp-0.5.1-orig/configure.ac	2016-07-07 05:00:16.000000000 +0400
+++ libwebp-0.5.1/configure.ac	2016-08-30 01:18:45.656307200 +0400
@@ -370,8 +370,41 @@
 
       # find libGL
       GL_SAVED_LIBS="$LIBS"
-      AC_SEARCH_LIBS([glBegin], [GL OpenGL opengl32])
+      _ac_test_glBegin_found=no
+      AC_SEARCH_LIBS([glBegin], [GL OpenGL opengl32], [_ac_test_glBegin_found=yes])
       LIBS="$GL_SAVED_LIBS"
+      if test "$_ac_test_glBegin_found" != "yes"; then
+        case $host_os in
+          mingw*)
+            AC_MSG_CHECKING(for Windows OpenGL support)
+            SAVED_LIBS=$LIBS
+            LIBS="-lopengl32 $LIBS"
+            opengl32_headers="
+              #include <windows.h>
+              #include <GL/gl.h>
+              "
+            opengl32_main="
+              int main(void) {
+                glBegin(0);
+                return 0;
+              }
+              "
+            AC_LANG_PUSH(C)
+            AC_LINK_IFELSE(
+              [AC_LANG_SOURCE([
+               $opengl32_headers
+               $opengl32_main])],
+              [opengl32_support=yes],
+              [opengl32_support=no]
+            )
+            AC_LANG_POP
+            LIBS=$SAVED_LIBS
+            AC_MSG_RESULT(${opengl32_support-no})
+
+            test "$opengl32_support" = "yes" && ac_cv_search_glBegin="-lopengl32"
+            ;;
+        esac
+      fi
 
       # A direct link to libGL may not be necessary on e.g., linux.
       GLUT_SAVED_LIBS="$LIBS"
@@ -400,6 +433,26 @@
           break
         fi
       done
+      if test "$opengl32_support" = "yes" -a "$glut_support" != "yes"; then
+        LIBS="-lglut -lopengl32 $LIBS"
+        AC_LINK_IFELSE(
+          [AC_LANG_PROGRAM([
+             #include <windows.h>
+             #include <GL/gl.h>
+             #include <GL/glut.h>
+            ],[
+             glOrtho(0, 0, 0, 0, 0, 0);
+             glutMainLoop();
+            ])
+          ],
+          AC_DEFINE(WEBP_HAVE_GL, [1],
+                    [Set to 1 if OpenGL is supported])
+          [glut_support=yes], []
+        )
+        if test "$glut_support" = "yes"; then
+          GL_LIBS="-lopengl32 -lglut"
+        fi
+      fi
       LIBS="$GLUT_SAVED_LIBS"
       LDFLAGS="$GLUT_SAVED_LDFLAGS"
       test "$glut_support" = "yes" && break
