diff -urBb freetype-2.4.11-orig/builds/unix/configure.ac freetype-2.4.11/builds/unix/configure.ac
--- freetype-2.4.11-orig/builds/unix/configure.ac	2012-12-20 11:35:16 +0400
+++ freetype-2.4.11/builds/unix/configure.ac	2014-08-07 22:24:16 +0400
@@ -215,7 +215,15 @@
 if test "x$GCC" = xyes; then
   XX_CFLAGS="-Wall"
   XX_ANSIFLAGS=""
-  for a in -pedantic -ansi
+  case $build in
+    *-*-mingw32 )
+      _ansi_flags="-pedantic -std=gnu99"
+      ;;
+    * )
+      _ansi_flags="-pedantic -ansi"
+      ;;
+  esac
+  for a in $_ansi_flags
   do
     AC_MSG_CHECKING([gcc compiler flag ${a} to assure ANSI C works correctly])
     orig_CFLAGS="${CFLAGS}"
