diff -urBb gcc-8.1.0-p/libiberty/Makefile.in gcc-8.1.0/libiberty/Makefile.in
--- gcc-8.1.0-p/libiberty/Makefile.in	2018-05-21 10:55:13.362591900 +0400
+++ gcc-8.1.0/libiberty/Makefile.in	2018-05-21 10:55:35.842972300 +0400
@@ -355,7 +355,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
