From https://bugreports.qt-project.org/browse/QTBUG-34856

diff -urBb qt-everywhere-opensource-src-4.8.5-orig/src/corelib/tools/qsimd.cpp qt-everywhere-opensource-src-4.8.5/src/corelib/tools/qsimd.cpp
--- qt-everywhere-opensource-src-4.8.5-orig/src/corelib/tools/qsimd.cpp	2013-06-07 09:16:52 +0400
+++ qt-everywhere-opensource-src-4.8.5/src/corelib/tools/qsimd.cpp	2014-03-24 14:16:31 +0400
@@ -47,7 +47,7 @@
 #include <windows.h>
 #endif
 
-#if defined(Q_OS_WIN64) && !defined(Q_CC_GNU)
+#if defined(Q_OS_WIN64) || defined(Q_OS_WIN32)
 #include <intrin.h>
 #endif
 
