diff -urBb wxWidgets-2.8.12/Makefile.in tmp/wxWidgets-build/wxWidgets-2.8.12/Makefile.in
--- wxWidgets-2.8.12/Makefile.in	2011-03-22 16:00:32 +0400
+++ tmp/wxWidgets-build/wxWidgets-2.8.12/Makefile.in	2013-01-12 21:46:50 +0400
@@ -12455,7 +12455,11 @@
 	$(INSTALL_DIR) $(DESTDIR)$(bindir)
 	$(INSTALL_DIR) $(DESTDIR)$(libdir)/wx/config
 	$(INSTALL_PROGRAM) lib/wx/config/$(TOOLCHAIN_FULLNAME) $(DESTDIR)$(libdir)/wx/config
+ifeq ($(TOOLKIT),MSW)
+	(cd $(DESTDIR)$(bindir) && rm -f wx-config && cp $(DESTDIR)$(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config)
+else
 	(cd $(DESTDIR)$(bindir) && rm -f wx-config && $(LN_S) $(libdir)/wx/config/$(TOOLCHAIN_FULLNAME) wx-config)
+endif
 
 locale_install: 
 	$(INSTALL_DIR) $(DESTDIR)$(datadir)/locale
