diff -urBbN giflib-5.2.1-orig/AUTHORS giflib-5.2.1/AUTHORS
--- giflib-5.2.1-orig/AUTHORS	1970-01-01 04:00:00.000000000 +0400
+++ giflib-5.2.1/AUTHORS	2014-05-16 14:46:53.000000000 +0400
@@ -0,0 +1,36 @@
+Michael Brown <michael_brown_uk[AT]hotmail.com>
+        callbacks to write data via user defined function
+
+Daniel Eisenbud <eisenbud[AT]google.com>
+	Fixes for crashes with invalid gif files and double freeing of
+	colormaps
+
+Gershon Elber <gershon[AT]cs.technion.sc.il>
+	original giflib code
+
+Marc Ewing <marc[AT]redhat.com>
+        spec file (for rpms) updates
+
+Toshio Kuratomi <toshio[AT]tiki-lounge.com>
+        uncompressed gif writing code
+        autoconf/automake process
+        former maintainer
+
+marek <marwaw[AT]users.sourceforge.net>
+	Gif initialization fix
+	windows build code
+
+Peter Mehlitz <peter[AT]transvirtual.com>
+        callbacks to read data from arbitrary sources (like libjpeg/libpng)
+
+Dick Porter <dick[AT]cymru.net>
+        int/pointer fixes for Alpha
+
+Eric Raymond <esr[AT]snark.thyrsus.com>
+        current as well as long time former maintainer of giflib code
+
+Petter Reinholdtsen <pere[AT]hungry.com>
+	Tru64 build fixs
+	
+Georg Schwarz <geos[AT]epost.de>
+	IRIX fixes
diff -urBbN giflib-5.2.1-orig/config.h.in giflib-5.2.1/config.h.in
--- giflib-5.2.1-orig/config.h.in	1970-01-01 04:00:00.000000000 +0400
+++ giflib-5.2.1/config.h.in	2020-02-02 09:42:22.316815300 +0400
@@ -0,0 +1,70 @@
+/* config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* Define to 1 if you have the <dlfcn.h> header file. */
+#undef HAVE_DLFCN_H
+
+/* Define to 1 if you have the <fcntl.h> header file. */
+#undef HAVE_FCNTL_H
+
+/* Define to 1 if you have the <inttypes.h> header file. */
+#undef HAVE_INTTYPES_H
+
+/* Define to 1 if you have the <limits.h> header file. */
+#undef HAVE_LIMITS_H
+
+/* Define to 1 if you have the <memory.h> header file. */
+#undef HAVE_MEMORY_H
+
+/* Define to 1 if you have the <stdarg.h> header file. */
+#undef HAVE_STDARG_H
+
+/* Define to 1 if you have the <stdint.h> header file. */
+#undef HAVE_STDINT_H
+
+/* Define to 1 if you have the <stdlib.h> header file. */
+#undef HAVE_STDLIB_H
+
+/* Define to 1 if you have the <strings.h> header file. */
+#undef HAVE_STRINGS_H
+
+/* Define to 1 if you have the <string.h> header file. */
+#undef HAVE_STRING_H
+
+/* Define to 1 if you have the <sys/stat.h> header file. */
+#undef HAVE_SYS_STAT_H
+
+/* Define to 1 if you have the <sys/types.h> header file. */
+#undef HAVE_SYS_TYPES_H
+
+/* Define to 1 if you have the <unistd.h> header file. */
+#undef HAVE_UNISTD_H
+
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
+#undef LT_OBJDIR
+
+/* Name of package */
+#undef PACKAGE
+
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Define to 1 if you have the ANSI C header files. */
+#undef STDC_HEADERS
+
+/* Version number of package */
+#undef VERSION
diff -urBbN giflib-5.2.1-orig/configure.ac giflib-5.2.1/configure.ac
--- giflib-5.2.1-orig/configure.ac	1970-01-01 04:00:00.000000000 +0400
+++ giflib-5.2.1/configure.ac	2020-02-02 09:12:10.040506800 +0400
@@ -0,0 +1,51 @@
+dnl Process this file with autoconf to produce a configure script.
+AC_INIT(giflib, [5.2.1], [esr@thyrsus.com], giflib)
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_SRCDIR([dgif_lib.c])
+AM_INIT_AUTOMAKE([gnu dist-bzip2 -Wall])
+dnl Make builds less verbose.  Shuts off Makefile portability checks.
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+
+dnl Note: config.h is not used in the current build
+dnl We leave this in place only to suppress an error message at autogen time
+AM_CONFIG_HEADER(config.h)
+
+dnl Checks for programs.
+AM_PROG_AR
+AC_PROG_LIBTOOL
+AC_PROG_CC
+AC_PROG_CPP
+AC_PROG_INSTALL
+AC_PROG_LN_S
+AC_PROG_MAKE_SET
+AM_PROG_CC_C_O
+
+dnl Allow partial building on systems without xmlto
+AC_CHECK_PROG([have_xmlto], [xmlto], ["yes"],["no"])
+AM_CONDITIONAL([BUILD_DOC], [test "x${have_xmlto}" = "xyes"])
+
+dnl Shared-library version
+SHLIBVERSION=7:2:0
+AC_SUBST(SHLIBVERSION)
+
+dnl Define warnings if we're compiling with gcc.
+dnl Set it in CFLAGS so the end-user can disable it if they wish.
+if test "x$GCC" = "xyes" ; then
+  CFLAGS="$CFLAGS -Wall"
+fi
+
+AC_CHECK_HEADERS(limits.h, ,
+                 AC_MSG_ERROR([giflib needs limits.h to compile]))
+AC_CHECK_HEADERS(fcntl.h, ,
+		 AC_MSG_ERROR([giflib needs fcntl.h to compile]))
+AC_CHECK_HEADERS(stdint.h, ,
+		 AC_MSG_ERROR([giflib needs stdint.h to compile]))
+AC_CHECK_HEADERS(stdarg.h, ,
+		 AC_MSG_ERROR([giflib needs stdarg.h to compile]))
+
+AC_CONFIG_FILES([
+Makefile
+doc/Makefile
+pic/Makefile
+])
+AC_OUTPUT
diff -urBbN giflib-5.2.1-orig/doc/Makefile.am giflib-5.2.1/doc/Makefile.am
--- giflib-5.2.1-orig/doc/Makefile.am	1970-01-01 04:00:00.000000000 +0400
+++ giflib-5.2.1/doc/Makefile.am	2020-02-02 09:10:25.730876200 +0400
@@ -0,0 +1,61 @@
+SUFFIXES = .xml .html .txt .asc .1
+
+.xml.html:
+	xmlto xhtml-nochunks $<
+
+.xml.1:
+	xmlto man $<
+
+.xml.txt:
+	xmlto txt $<
+
+.asc.html:
+	asciidoc $<
+
+all: allhtml manpages
+
+# The distinction between XMLMAN and XMLINTERNAL is because
+# some pages shouldn't be installed as part of a binary package;
+# they're just for test-pattern generators.
+XMLMAN = \
+	gif2rgb.xml \
+	gifbuild.xml \
+	gifclrmp.xml \
+	gifecho.xml \
+	giffix.xml \
+	gifinto.xml \
+	giflib.xml \
+	giftext.xml \
+	giftool.xml
+XMLINTERNAL = \
+	gifbg.xml \
+	gifcolor.xml \
+	gifhisto.xml \
+	gifwedge.xml
+XMLDOC = intro.xml gif_lib.xml
+XMLALL = $(XMLMAN) $(XMLINTERNAL) $(XMLDOC)
+TXTDOC = 00README
+SOURCES = $(XMLALL) $(TXTDOC)
+SITEHTML = index.html
+
+# Philosophical choice: the website gets the internal manual pages
+allhtml: $(XMLALL:.xml=.html) $(XMLINTERNAL:.xml=.html)
+
+manpages: $(XMLMAN:.xml=.1) $(XMLINTERNAL:.xml=.1)
+man_MANS = $(XMLMAN:.xml=.1)
+
+# Control what goes in the distribution tarball.
+# We include all of the XML, and also generated manual pages 
+# so people working from the distribution tarball won't need xmlto. 
+EXTRA_DIST = $(SOURCES) $(XMLMAN:.xml=.1) $(XMLINTERNAL:.xml=.1)
+
+# Prepare the website directory to deliver an update.
+# ImageMagick and asciidoc are required.
+website: allhtml
+	rm -fr staging; mkdir staging; 
+	cp -r $(XMLALL:.xml=.html) $(SITEHTML) *.txt whatsinagif staging
+	convert ../pic/gifgrid.gif -resize 50x50 staging/giflib-logo.gif
+	asciidoc - <../history.asc >staging/history.html
+
+clean-local:
+	rm -fr staging $(XMLALL:.xml=.html) $(XMLMAN:.xml=.1)
diff -urBbN giflib-5.2.1-orig/Makefile.am giflib-5.2.1/Makefile.am
--- giflib-5.2.1-orig/Makefile.am	1970-01-01 04:00:00.000000000 +0400
+++ giflib-5.2.1/Makefile.am	2020-02-02 09:54:15.340774600 +0400
@@ -0,0 +1,110 @@
+SUBDIRS = pic
+
+# Avoid trouble if your system doesn't have xmlto.
+if BUILD_DOC
+SUBDIRS += doc
+endif
+
+EXTRA_DIST = autogen.sh \
+         tests \
+         TODO \
+         build.adoc \
+         history.adoc \
+         control \
+         doc/whatsinagif
+
+# library
+lib_LTLIBRARIES = libgif.la
+
+include_HEADERS = gif_lib.h
+
+libgif_la_SOURCES = dgif_lib.c  \
+                    egif_lib.c  \
+                    gif_font.c  \
+                    gif_hash.c  \
+                    gif_hash.h	\
+                    gifalloc.c  \
+                    openbsd-reallocarray.c \
+                    gif_err.c   \
+                    gif_lib_private.h \
+                    quantize.c
+
+libgif_la_LDFLAGS = -no-undefined -version-info @SHLIBVERSION@
+libgif_CFLAGS = $(X_CFLAGS) $(AM_CFLAGS)
+
+# Utilities
+
+# The no-install programs are test pattern generators and example code.
+noinst_PROGRAMS = gifbg gifcolor giffilter gifhisto gifsponge gifwedge
+
+noinst_LIBRARIES = libgetarg.a
+
+bin_PROGRAMS = \
+	gif2rgb \
+	gifbuild \
+	gifecho \
+	giffix \
+	gifinto \
+	giftext \
+	giftool \
+	gifclrmp
+
+gif2rgb_LDADD = libgetarg.a libgif.la
+gifbuild_LDADD = libgetarg.a libgif.la
+gifecho_LDADD = libgetarg.a libgif.la
+giffix_LDADD = libgetarg.a libgif.la
+gifinto_LDADD = libgetarg.a libgif.la
+giftext_LDADD = libgetarg.a libgif.la
+giftool_LDADD = libgetarg.a libgif.la
+gifclrmp_LDADD = libgetarg.a libgif.la
+
+# non installable utils
+gifbg_LDADD = libgetarg.a libgif.la
+gifcolor_LDADD = libgetarg.a libgif.la
+giffilter_LDADD = libgetarg.a libgif.la
+gifhisto_LDADD = libgetarg.a libgif.la
+gifsponge_LDADD = libgetarg.a libgif.la
+gifwedge_LDADD = libgetarg.a libgif.la
+
+libgetarg_a_SOURCES = getarg.c getarg.h \
+                    qprintf.c
+
+gifclrmp_SOURCES = gifclrmp.c
+gifclrmp_LDADD += -lm
+
+gif2rgb_SOURCES = gif2rgb.c
+gifbg_SOURCES = gifbg.c
+gifbuild_SOURCES = gifbuild.c
+gifcolor_SOURCES = gifcolor.c
+gifecho_SOURCES = gifecho.c
+giffilter_SOURCES = giffilter.c
+giffix_SOURCES = giffix.c
+gifhisto_SOURCES = gifhisto.c
+gifinto_SOURCES = gifinto.c
+gifsponge_SOURCES = gifsponge.c
+giftext_SOURCES = giftext.c
+giftool_SOURCES = giftool.c
+gifwedge_SOURCES = gifwedge.c
+
+# Auditing tools.
+
+# cppcheck should run clean
+cppcheck:
+	cppcheck -Ilib --inline-suppr --template gcc --enable=all --suppress=unusedStructMember --suppress=unusedFunction --force lib/*.[ch] util/*.[ch]
+
+# splint is nowhere near running clean
+SPLINT_SUPPRESSIONS = -boolops -nullret
+splint: 
+	splint +quiet +posixlib $(SPLINT_SUPPRESSIONS) lib/*.[ch]
+
+# release using the shipper tool, if distcheck passes
+release: distcheck
+	cd doc; make website
+	shipper version=@VERSION@ | sh -e -x
+	rm -fr doc/staging
+
+# Refresh the wbsite
+refresh:
+	cd doc; make website
+	shipper -w version=@VERSION@ | sh -e -x
+	rm -fr doc/staging
diff -urBbN giflib-5.2.1-orig/pic/Makefile.am giflib-5.2.1/pic/Makefile.am
--- giflib-5.2.1-orig/pic/Makefile.am	1970-01-01 04:00:00.000000000 +0400
+++ giflib-5.2.1/pic/Makefile.am	2014-05-16 14:46:53.000000000 +0400
@@ -0,0 +1,11 @@
+EXTRA_DIST = NOTES \
+	     fire.gif \
+	     gifgrid.gif \
+	     porsche.gif \
+	     solid2.gif \
+	     treescap.gif \
+	     treescap-interlaced.gif \
+	     welcome2.gif \
+	     x-trans.gif \
+	     sample.ico
+
