diff -urBb gcc-9.1.0-orig/libatomic/configure gcc-9.1.0/libatomic/configure
--- gcc-9.1.0-orig/libatomic/configure	2019-05-03 12:01:59.000000000 +0400
+++ gcc-9.1.0/libatomic/configure	2019-05-13 09:17:07.296646500 +0400
@@ -15694,7 +15694,7 @@
   multilib_arg=
 fi
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
+ac_config_files="$ac_config_files Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -16766,7 +16766,6 @@
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "gstdint.h") CONFIG_COMMANDS="$CONFIG_COMMANDS gstdint.h" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-9.1.0-orig/libatomic/configure.ac gcc-9.1.0/libatomic/configure.ac
--- gcc-9.1.0-orig/libatomic/configure.ac	2019-01-01 16:31:55.000000000 +0400
+++ gcc-9.1.0/libatomic/configure.ac	2019-05-13 09:17:07.301646100 +0400
@@ -278,5 +278,5 @@
   multilib_arg=
 fi
 
-AC_CONFIG_FILES(Makefile testsuite/Makefile)
+AC_CONFIG_FILES(Makefile)
 AC_OUTPUT
diff -urBb gcc-9.1.0-orig/libatomic/Makefile.am gcc-9.1.0/libatomic/Makefile.am
--- gcc-9.1.0-orig/libatomic/Makefile.am	2019-01-01 16:31:55.000000000 +0400
+++ gcc-9.1.0/libatomic/Makefile.am	2019-05-13 09:17:07.303645400 +0400
@@ -23,7 +23,7 @@
 ##  <http://www.gnu.org/licenses/>.
 
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-9.1.0-orig/libatomic/Makefile.in gcc-9.1.0/libatomic/Makefile.in
--- gcc-9.1.0-orig/libatomic/Makefile.in	2019-05-03 12:01:59.000000000 +0400
+++ gcc-9.1.0/libatomic/Makefile.in	2019-05-13 09:17:07.304644800 +0400
@@ -381,7 +381,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) \
 	$(top_srcdir) $(top_builddir)
diff -urBb gcc-9.1.0-orig/libgomp/configure gcc-9.1.0/libgomp/configure
--- gcc-9.1.0-orig/libgomp/configure	2019-05-03 12:01:59.000000000 +0400
+++ gcc-9.1.0/libgomp/configure	2019-05-13 09:17:07.312642200 +0400
@@ -17260,9 +17260,9 @@
 
 ac_config_files="$ac_config_files omp.h omp_lib.h omp_lib.f90 libgomp_f.h"
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile libgomp.spec"
+ac_config_files="$ac_config_files Makefile libgomp.spec"
 
-ac_config_files="$ac_config_files testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in"
+ac_config_files="$ac_config_files"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
diff -urBb gcc-9.1.0-orig/libgomp/configure.ac gcc-9.1.0/libgomp/configure.ac
--- gcc-9.1.0-orig/libgomp/configure.ac	2019-05-03 12:01:59.000000000 +0400
+++ gcc-9.1.0/libgomp/configure.ac	2019-05-13 09:17:07.317639300 +0400
@@ -423,6 +423,5 @@
 GCC_BASE_VER
 
 AC_CONFIG_FILES(omp.h omp_lib.h omp_lib.f90 libgomp_f.h)
-AC_CONFIG_FILES(Makefile testsuite/Makefile libgomp.spec)
-AC_CONFIG_FILES([testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in])
+AC_CONFIG_FILES(Makefile libgomp.spec)
 AC_OUTPUT
diff -urBb gcc-9.1.0-orig/libgomp/Makefile.am gcc-9.1.0/libgomp/Makefile.am
--- gcc-9.1.0-orig/libgomp/Makefile.am	2019-05-03 12:01:59.000000000 +0400
+++ gcc-9.1.0/libgomp/Makefile.am	2019-05-13 09:17:07.318639000 +0400
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = info-in-builddir
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-9.1.0-orig/libgomp/Makefile.in gcc-9.1.0/libgomp/Makefile.in
--- gcc-9.1.0-orig/libgomp/Makefile.in	2019-05-03 12:01:59.000000000 +0400
+++ gcc-9.1.0/libgomp/Makefile.in	2019-05-13 09:17:07.320638300 +0400
@@ -516,7 +516,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = info-in-builddir
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir) \
 	      $(top_srcdir)/../include
diff -urBb gcc-9.1.0-orig/libiberty/configure gcc-9.1.0/libiberty/configure
--- gcc-9.1.0-orig/libiberty/configure	2018-10-31 21:03:16.000000000 +0400
+++ gcc-9.1.0/libiberty/configure	2019-05-13 09:17:07.324637100 +0400
@@ -7471,7 +7471,7 @@
 
 
 # We need multilib support, but only if configuring for the target.
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
+ac_config_files="$ac_config_files Makefile"
 
 ac_config_commands="$ac_config_commands default"
 
diff -urBb gcc-9.1.0-orig/libiberty/configure.ac gcc-9.1.0/libiberty/configure.ac
--- gcc-9.1.0-orig/libiberty/configure.ac	2018-10-31 21:03:16.000000000 +0400
+++ gcc-9.1.0/libiberty/configure.ac	2019-05-13 09:17:07.327634900 +0400
@@ -728,7 +728,7 @@
 AC_SUBST(htmldir)
 
 # We need multilib support, but only if configuring for the target.
-AC_CONFIG_FILES([Makefile testsuite/Makefile])
+AC_CONFIG_FILES([Makefile])
 AC_CONFIG_COMMANDS([default],
   [[test -z "$CONFIG_HEADERS" || echo timestamp > stamp-h
 if test -n "$CONFIG_FILES"; then
diff -urBb gcc-9.1.0-orig/libiberty/Makefile.in gcc-9.1.0/libiberty/Makefile.in
--- gcc-9.1.0-orig/libiberty/Makefile.in	2019-05-13 09:16:45.792599900 +0400
+++ gcc-9.1.0/libiberty/Makefile.in	2019-05-13 09:17:07.329634500 +0400
@@ -97,7 +97,7 @@
 	"tooldir=$(tooldir)"
 
 # Subdirectories to recurse into. We need to override this during cleaning
-SUBDIRS = testsuite
+SUBDIRS = 
 
 # FIXME: add @BUILD_INFO@ once we're sure it works for everyone.
 all: stamp-picdir stamp-noasandir $(TARGETLIB) required-list all-subdir
diff -urBb gcc-9.1.0-orig/libstdc++-v3/configure gcc-9.1.0/libstdc++-v3/configure
--- gcc-9.1.0-orig/libstdc++-v3/configure	2019-03-11 20:28:11.000000000 +0400
+++ gcc-9.1.0/libstdc++-v3/configure	2019-05-13 09:19:57.672638600 +0400
@@ -5204,7 +5204,7 @@
   # expandable list at autoconf time; the second provides an expandable list
   # (i.e., shell variable) at configure time.
 
-  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/c++17 src/filesystem doc po testsuite python'
+  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/c++17 src/filesystem doc po python'
 
   # These need to be absolute paths, yet at the same time need to
   # canonicalize only relative paths, because then amd will not unmount
@@ -82419,7 +82419,7 @@
 # append it here.  Only modify Makefiles that have just been created.
 #
 # Also, get rid of this simulated-VPATH thing that automake does.
-ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/c++17/Makefile src/filesystem/Makefile doc/Makefile po/Makefile testsuite/Makefile python/Makefile"
+ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/c++17/Makefile src/filesystem/Makefile doc/Makefile po/Makefile python/Makefile"
 
 
 ac_config_commands="$ac_config_commands generate-headers"
diff -urBb gcc-9.1.0-orig/libstdc++-v3/Makefile.am gcc-9.1.0/libstdc++-v3/Makefile.am
--- gcc-9.1.0-orig/libstdc++-v3/Makefile.am	2019-01-01 16:31:55.000000000 +0400
+++ gcc-9.1.0/libstdc++-v3/Makefile.am	2019-05-13 09:17:07.358626700 +0400
@@ -24,7 +24,7 @@
 
 if GLIBCXX_HOSTED
 ## Note that python must come after src.
-  hosted_source = src doc po testsuite python
+  hosted_source = src doc po python
 endif
 
 ## Keep this list sync'd with acinclude.m4:GLIBCXX_CONFIGURE.
diff -urBb gcc-9.1.0-orig/libstdc++-v3/Makefile.in gcc-9.1.0/libstdc++-v3/Makefile.in
--- gcc-9.1.0-orig/libstdc++-v3/Makefile.in	2019-02-27 15:25:44.000000000 +0400
+++ gcc-9.1.0/libstdc++-v3/Makefile.in	2019-05-13 09:21:07.743605400 +0400
@@ -179,7 +179,7 @@
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = include libsupc++ src doc po testsuite python
+DIST_SUBDIRS = include libsupc++ src doc po python
 ABI_TWEAKS_SRCDIR = @ABI_TWEAKS_SRCDIR@
 ACLOCAL = @ACLOCAL@
 ALLOCATOR_H = @ALLOCATOR_H@
@@ -406,7 +406,7 @@
 
 # -I/-D flags to pass when compiling.
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES) $(CPPFLAGS)
-@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po testsuite python
+@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po python
 SUBDIRS = include libsupc++ $(hosted_source)
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 @BUILD_XML_FALSE@STAMP_XML = 
diff -urBb gcc-9.1.0-orig/libvtv/configure gcc-9.1.0/libvtv/configure
--- gcc-9.1.0-orig/libvtv/configure	2018-10-31 21:03:16.000000000 +0400
+++ gcc-9.1.0/libvtv/configure	2019-05-13 09:17:07.366622900 +0400
@@ -15715,11 +15715,6 @@
 ac_config_files="$ac_config_files Makefile"
 
 
-if test "x$VTV_SUPPORTED" = "xyes"; then
-  ac_config_files="$ac_config_files testsuite/Makefile"
-
-fi
-
 case "$target_os" in
   cygwin*|mingw32*)
     vtv_cygmin="yes"
diff -urBb gcc-9.1.0-orig/libvtv/configure.ac gcc-9.1.0/libvtv/configure.ac
--- gcc-9.1.0-orig/libvtv/configure.ac	2018-10-31 21:03:16.000000000 +0400
+++ gcc-9.1.0/libvtv/configure.ac	2019-05-13 09:17:07.371621200 +0400
@@ -165,21 +165,6 @@
 
 AC_CONFIG_FILES([Makefile])
 
-if test "x$VTV_SUPPORTED" = "xyes"; then
-  AC_CONFIG_FILES(AC_FOREACH([DIR], [testsuite], [DIR/Makefile ]), 
-    [cat > vpsed$$ << \_EOF
-s!`test -f '$<' || echo '$(srcdir)/'`!!
-_EOF
-    sed -f vpsed$$ $ac_file > tmp$$
-    mv tmp$$ $ac_file
-    rm vpsed$$
-    echo 'MULTISUBDIR =' >> $ac_file
-    ml_norecursion=yes
-    . ${multi_basedir}/config-ml.in
-    AS_UNSET([ml_norecursion])
-])
-fi
-
 case "$target_os" in
   cygwin*|mingw32*)
     vtv_cygmin="yes"
