diff -urBb gcc-8.1.0-orig/libstdc++-v3/config/abi/pre/gnu.ver gcc-8.1.0/libstdc++-v3/config/abi/pre/gnu.ver
--- gcc-8.1.0-orig/libstdc++-v3/config/abi/pre/gnu.ver	2018-01-16 16:43:08.000000000 +0400
+++ gcc-8.1.0/libstdc++-v3/config/abi/pre/gnu.ver	2018-05-21 10:30:09.880787400 +0400
@@ -2178,7 +2178,7 @@
     __cxa_allocate_dependent_exception;
     __cxa_free_dependent_exception;
     __cxa_deleted_virtual;
-
+    _ZTC*;
 } CXXABI_1.3.5;
 
 CXXABI_1.3.7 {
