diff -urBb gcc-7.1.0-orig/gcc/config.gcc gcc-7.1.0/gcc/config.gcc
--- gcc-7.1.0-orig/gcc/config.gcc	2017-03-24 13:59:51.000000000 +0000
+++ gcc-7.1.0/gcc/config.gcc	2017-05-09 10:59:05.664180500 +0000
@@ -1758,7 +1758,10 @@
 			tmake_file="${tmake_file} i386/t-mingw-w32"
 			;;
 	esac
+	# Don't ignore values passed in to configure via --native-system-header-dir
+	if test x$native_system_header_dir = x ; then
         native_system_header_dir=/mingw/include
+	fi
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	extra_options="${extra_options} i386/cygming.opt i386/mingw.opt"
 	case ${target} in
