diff -urBb gcc-7.1.0-orig/libgcc/gthr-posix.h gcc-7.1.0/libgcc/gthr-posix.h
--- gcc-7.1.0-orig/libgcc/gthr-posix.h	2017-01-01 12:07:43.000000000 +0000
+++ gcc-7.1.0/libgcc/gthr-posix.h	2017-05-09 11:05:24.637830800 +0000
@@ -82,7 +82,7 @@
 # define __GTHREAD_COND_INIT_FUNCTION __gthread_cond_init_function
 #endif
 
-#if SUPPORTS_WEAK && GTHREAD_USE_WEAK
+#if SUPPORTS_WEAK && GTHREAD_USE_WEAK && !defined(__MINGW32__)
 # ifndef __gthrw_pragma
 #  define __gthrw_pragma(pragma)
 # endif
