diff -urBbN gcc-6.3.0-orig/Makefile.in gcc-6.3.0/Makefile.in
--- gcc-6.3.0-orig/Makefile.in	2016-04-13 22:02:08.000000000 +0400
+++ gcc-6.3.0/Makefile.in	2016-12-23 20:46:05.636182600 +0400
@@ -245,7 +245,8 @@
 	  `if $(LEAN); then echo ' -isystem '; else echo ' -I'; fi`$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/include \
 	  `if $(LEAN); then echo ' -isystem '; else echo ' -I'; fi`$$s/libstdc++-v3/libsupc++ \
 	  -L$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/src/.libs \
-	  -L$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/libsupc++/.libs"; \
+	  -L$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/libsupc++/.libs \
+	  $(XGCC_FLAGS_FOR_TARGET) $$TFLAGS"; \
 	  export CXX; \
 	CXX_FOR_BUILD="$$CXX"; export CXX_FOR_BUILD;
 @endif target-libstdc++-v3-bootstrap
