diff -urBb gcc-6.2.0-orig/libstdc++-v3/config/abi/pre/gnu.ver gcc-6.2.0/libstdc++-v3/config/abi/pre/gnu.ver
--- gcc-6.2.0-orig/libstdc++-v3/config/abi/pre/gnu.ver	2016-03-24 22:13:40.000000000 +0400
+++ gcc-6.2.0/libstdc++-v3/config/abi/pre/gnu.ver	2016-08-28 11:38:36.872000000 +0400
@@ -2101,7 +2101,7 @@
     __cxa_allocate_dependent_exception;
     __cxa_free_dependent_exception;
     __cxa_deleted_virtual;
-
+    _ZTC*;
 } CXXABI_1.3.5;
 
 CXXABI_1.3.7 {
