diff -urBb gcc-15.2.0-orig/configure gcc-15.2.0/configure
--- gcc-15.2.0-orig/configure	2025-08-08 10:51:39.874338100 +0400
+++ gcc-15.2.0/configure	2026-01-11 12:32:25.370829000 +0400
@@ -11578,7 +11578,7 @@
   x86_64-*mingw* | *-w64-mingw*)
   # MinGW-w64 does not use newlib, nor does it use winsup. It may,
   # however, use a symlink named 'mingw' in ${prefix} .
-    FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/mingw/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/mingw/include'
+    FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/include'
     ;;
   *-mingw*)
   # MinGW can't be handled as Cygwin above since it does not use newlib.
diff -urBb gcc-15.2.0-orig/configure.ac gcc-15.2.0/configure.ac
--- gcc-15.2.0-orig/configure.ac	2025-08-08 10:51:39.874338100 +0400
+++ gcc-15.2.0/configure.ac	2026-01-11 12:32:25.374853300 +0400
@@ -3805,7 +3805,7 @@
   x86_64-*mingw* | *-w64-mingw*)
   # MinGW-w64 does not use newlib, nor does it use winsup. It may,
   # however, use a symlink named 'mingw' in ${prefix} .
-    FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/mingw/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/mingw/include'
+    FLAGS_FOR_TARGET=$FLAGS_FOR_TARGET' -L${prefix}/${target}/lib -L${prefix}/lib -isystem ${prefix}/${target}/include -isystem ${prefix}/include'
     ;;
   *-mingw*)
   # MinGW can't be handled as Cygwin above since it does not use newlib.
