diff -urBb gcc-6.2.0-orig/libiberty/Makefile.in gcc-6.2.0/libiberty/Makefile.in
--- gcc-6.2.0-orig/libiberty/Makefile.in	2015-11-12 23:18:54.000000000 +0400
+++ gcc-6.2.0/libiberty/Makefile.in	2016-08-28 11:30:52.755372000 +0400
@@ -355,7 +355,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
