diff -urBb ../../gcc-4.7.1-orig/gcc/Makefile.in gcc-4.7.1/gcc/Makefile.in
--- ../../gcc-4.7.1-orig/gcc/Makefile.in	2012-05-18 20:32:05 +0400
+++ gcc-4.7.1/gcc/Makefile.in	2012-08-23 09:59:15 +0400
@@ -134,7 +134,7 @@
 T_CFLAGS =
 TCFLAGS =
 CFLAGS = @CFLAGS@
-CXXFLAGS = @CXXFLAGS@
+CXXFLAGS = @CXXFLAGS@ -fpermissive
 LDFLAGS = @LDFLAGS@
 
 # Flags to determine code coverage. When coverage is disabled, this will
@@ -155,7 +155,7 @@
 # apply to the back end and some front ends, which may be compiled
 # with other compilers.
 # C_STRICT_WARN is similar, with C-only warnings.
-LOOSE_WARN = @loose_warn@
+LOOSE_WARN = @loose_warn@ -fpermissive
 C_LOOSE_WARN = @c_loose_warn@
 STRICT_WARN = @strict_warn@
 C_STRICT_WARN = @c_strict_warn@
