--- gcc-5.1.0-orig/gcc/config.gcc	2015-03-10 12:50:41.000000000 +0300
+++ gcc-5.1.0/gcc/config.gcc	2015-06-14 13:17:37.060042100 +0300
@@ -1693,7 +1693,10 @@
 			tmake_file="${tmake_file} i386/t-mingw-w32"
 			;;
 	esac
-        native_system_header_dir=/mingw/include
+	# Don't ignore values passed in to configure via --native-system-header-dir
+	if test x$native_system_header_dir = x ; then
+		native_system_header_dir=/mingw/include
+	fi
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	extra_options="${extra_options} i386/cygming.opt i386/mingw.opt"
 	case ${target} in
