diff -urBb gcc-11.2.0-orig/libatomic/Makefile.am gcc-11.2.0/libatomic/Makefile.am
--- gcc-11.2.0-orig/libatomic/Makefile.am	2021-07-28 10:55:08.692306900 +0400
+++ gcc-11.2.0/libatomic/Makefile.am	2021-10-12 11:00:59.491122300 +0400
@@ -23,7 +23,7 @@
 ##  <http://www.gnu.org/licenses/>.
 
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-11.2.0-orig/libatomic/Makefile.in gcc-11.2.0/libatomic/Makefile.in
--- gcc-11.2.0-orig/libatomic/Makefile.in	2021-07-28 10:55:08.696307000 +0400
+++ gcc-11.2.0/libatomic/Makefile.in	2021-10-12 11:00:59.493118400 +0400
@@ -383,7 +383,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) \
 	$(top_srcdir) $(top_builddir)
diff -urBb gcc-11.2.0-orig/libatomic/configure gcc-11.2.0/libatomic/configure
--- gcc-11.2.0-orig/libatomic/configure	2021-07-28 10:55:08.696307000 +0400
+++ gcc-11.2.0/libatomic/configure	2021-10-12 11:00:59.483146600 +0400
@@ -15736,9 +15736,7 @@
   multilib_arg=
 fi
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
-
-ac_config_files="$ac_config_files testsuite/libatomic-site-extra.exp"
+ac_config_files="$ac_config_files Makefile"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -16810,8 +16808,6 @@
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "gstdint.h") CONFIG_COMMANDS="$CONFIG_COMMANDS gstdint.h" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
-    "testsuite/libatomic-site-extra.exp") CONFIG_FILES="$CONFIG_FILES testsuite/libatomic-site-extra.exp" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-11.2.0-orig/libatomic/configure.ac gcc-11.2.0/libatomic/configure.ac
--- gcc-11.2.0-orig/libatomic/configure.ac	2021-07-28 10:55:08.696307000 +0400
+++ gcc-11.2.0/libatomic/configure.ac	2021-10-12 11:00:59.489127300 +0400
@@ -300,6 +300,5 @@
   multilib_arg=
 fi
 
-AC_CONFIG_FILES(Makefile testsuite/Makefile)
-AC_CONFIG_FILES(testsuite/libatomic-site-extra.exp)
+AC_CONFIG_FILES(Makefile)
 AC_OUTPUT
diff -urBb gcc-11.2.0-orig/libgomp/Makefile.am gcc-11.2.0/libgomp/Makefile.am
--- gcc-11.2.0-orig/libgomp/Makefile.am	2021-07-28 10:55:43.752796100 +0400
+++ gcc-11.2.0/libgomp/Makefile.am	2021-10-12 11:00:59.507083300 +0400
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = info-in-builddir
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-11.2.0-orig/libgomp/Makefile.in gcc-11.2.0/libgomp/Makefile.in
--- gcc-11.2.0-orig/libgomp/Makefile.in	2021-07-28 10:55:43.752796100 +0400
+++ gcc-11.2.0/libgomp/Makefile.in	2021-10-12 11:00:59.507083300 +0400
@@ -523,7 +523,7 @@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = info-in-builddir
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell @get_gcc_base_ver@ $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir) \
 	      $(top_srcdir)/../include
diff -urBb gcc-11.2.0-orig/libgomp/configure gcc-11.2.0/libgomp/configure
--- gcc-11.2.0-orig/libgomp/configure	2021-07-28 10:55:43.756796200 +0400
+++ gcc-11.2.0/libgomp/configure	2021-10-12 11:00:59.499072200 +0400
@@ -17044,11 +17044,9 @@
 
 ac_config_files="$ac_config_files omp.h omp_lib.h omp_lib.f90 libgomp_f.h"
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile libgomp.spec"
+ac_config_files="$ac_config_files Makefile libgomp.spec"
 
-ac_config_files="$ac_config_files testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in"
-
-ac_config_files="$ac_config_files testsuite/libgomp-site-extra.exp"
+ac_config_files="$ac_config_files"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
@@ -18196,10 +18194,7 @@
     "omp_lib.f90") CONFIG_FILES="$CONFIG_FILES omp_lib.f90" ;;
     "libgomp_f.h") CONFIG_FILES="$CONFIG_FILES libgomp_f.h" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
     "libgomp.spec") CONFIG_FILES="$CONFIG_FILES libgomp.spec" ;;
-    "testsuite/libgomp-test-support.pt.exp") CONFIG_FILES="$CONFIG_FILES testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in" ;;
-    "testsuite/libgomp-site-extra.exp") CONFIG_FILES="$CONFIG_FILES testsuite/libgomp-site-extra.exp" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-11.2.0-orig/libgomp/configure.ac gcc-11.2.0/libgomp/configure.ac
--- gcc-11.2.0-orig/libgomp/configure.ac	2021-07-28 10:55:09.056312000 +0400
+++ gcc-11.2.0/libgomp/configure.ac	2021-10-12 11:00:59.503061600 +0400
@@ -463,7 +463,5 @@
 GCC_BASE_VER
 
 AC_CONFIG_FILES(omp.h omp_lib.h omp_lib.f90 libgomp_f.h)
-AC_CONFIG_FILES(Makefile testsuite/Makefile libgomp.spec)
-AC_CONFIG_FILES([testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in])
-AC_CONFIG_FILES([testsuite/libgomp-site-extra.exp])
+AC_CONFIG_FILES(Makefile libgomp.spec)
 AC_OUTPUT
diff -urBb gcc-11.2.0-orig/libiberty/Makefile.in gcc-11.2.0/libiberty/Makefile.in
--- gcc-11.2.0-orig/libiberty/Makefile.in	2021-10-12 10:52:05.272499200 +0400
+++ gcc-11.2.0/libiberty/Makefile.in	2021-10-12 11:00:59.517052400 +0400
@@ -97,7 +97,7 @@
 	"tooldir=$(tooldir)"
 
 # Subdirectories to recurse into. We need to override this during cleaning
-SUBDIRS = testsuite
+SUBDIRS = 
 
 # FIXME: add @BUILD_INFO@ once we're sure it works for everyone.
 all: stamp-picdir stamp-noasandir $(TARGETLIB) required-list all-subdir
diff -urBb gcc-11.2.0-orig/libiberty/configure gcc-11.2.0/libiberty/configure
--- gcc-11.2.0-orig/libiberty/configure	2021-07-28 10:55:09.092312500 +0400
+++ gcc-11.2.0/libiberty/configure	2021-10-12 11:00:59.511068900 +0400
@@ -7661,7 +7661,7 @@
 
 
 # We need multilib support, but only if configuring for the target.
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
+ac_config_files="$ac_config_files Makefile"
 
 ac_config_commands="$ac_config_commands default"
 
@@ -8376,7 +8376,6 @@
   case $ac_config_target in
     "config.h") CONFIG_HEADERS="$CONFIG_HEADERS config.h:config.in" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
     "default") CONFIG_COMMANDS="$CONFIG_COMMANDS default" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
diff -urBb gcc-11.2.0-orig/libiberty/configure.ac gcc-11.2.0/libiberty/configure.ac
--- gcc-11.2.0-orig/libiberty/configure.ac	2021-07-28 10:55:09.092312500 +0400
+++ gcc-11.2.0/libiberty/configure.ac	2021-10-12 11:00:59.515030400 +0400
@@ -732,7 +732,7 @@
 AC_SUBST(htmldir)
 
 # We need multilib support, but only if configuring for the target.
-AC_CONFIG_FILES([Makefile testsuite/Makefile])
+AC_CONFIG_FILES([Makefile])
 AC_CONFIG_COMMANDS([default],
   [[test -z "$CONFIG_HEADERS" || echo timestamp > stamp-h
 if test -n "$CONFIG_FILES"; then
diff -urBb gcc-11.2.0-orig/libstdc++-v3/Makefile.am gcc-11.2.0/libstdc++-v3/Makefile.am
--- gcc-11.2.0-orig/libstdc++-v3/Makefile.am	2021-07-28 10:55:09.208314100 +0400
+++ gcc-11.2.0/libstdc++-v3/Makefile.am	2021-10-12 11:00:59.554951400 +0400
@@ -24,7 +24,7 @@
 
 if GLIBCXX_HOSTED
 ## Note that python must come after src.
-  hosted_source = src doc po testsuite python
+  hosted_source = src doc po python
 endif
 
 ## Keep this list sync'd with acinclude.m4:GLIBCXX_CONFIGURE.
@@ -32,10 +32,6 @@
 
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 
-# Testsuite/check forwarding targets.
-check-%:
-	cd testsuite && $(MAKE) $@
-
 # Documentation forwarding targets.
 doc-%:
 	cd doc && $(MAKE) $@
diff -urBb gcc-11.2.0-orig/libstdc++-v3/Makefile.in gcc-11.2.0/libstdc++-v3/Makefile.in
--- gcc-11.2.0-orig/libstdc++-v3/Makefile.in	2021-07-28 10:55:09.208314100 +0400
+++ gcc-11.2.0/libstdc++-v3/Makefile.in	2021-10-12 11:00:59.556948400 +0400
@@ -180,7 +180,7 @@
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = include libsupc++ src doc po testsuite python
+DIST_SUBDIRS = include libsupc++ src doc po python
 ABI_TWEAKS_SRCDIR = @ABI_TWEAKS_SRCDIR@
 ACLOCAL = @ACLOCAL@
 ALLOCATOR_H = @ALLOCATOR_H@
@@ -412,7 +412,7 @@
 
 # -I/-D flags to pass when compiling.
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES) $(CPPFLAGS)
-@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po testsuite python
+@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po python
 SUBDIRS = include libsupc++ $(hosted_source)
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 @BUILD_XML_FALSE@STAMP_XML = 
@@ -767,10 +767,6 @@
 .PRECIOUS: Makefile
 
 
-# Testsuite/check forwarding targets.
-check-%:
-	cd testsuite && $(MAKE) $@
-
 # Documentation forwarding targets.
 doc-%:
 	cd doc && $(MAKE) $@
diff -urBb gcc-11.2.0-orig/libstdc++-v3/configure gcc-11.2.0/libstdc++-v3/configure
--- gcc-11.2.0-orig/libstdc++-v3/configure	2021-07-28 10:55:09.228314400 +0400
+++ gcc-11.2.0/libstdc++-v3/configure	2021-10-12 11:05:08.307180500 +0400
@@ -5492,7 +5492,7 @@
   # expandable list at autoconf time; the second provides an expandable list
   # (i.e., shell variable) at configure time.
 
-  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/c++17 src/c++20 src/filesystem doc po testsuite python'
+  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/c++17 src/c++20 src/filesystem doc po python'
 
   # These need to be absolute paths, yet at the same time need to
   # canonicalize only relative paths, because then amd will not unmount
@@ -78007,7 +78007,7 @@
 # append it here.  Only modify Makefiles that have just been created.
 #
 # Also, get rid of this simulated-VPATH thing that automake does.
-ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/c++17/Makefile src/c++20/Makefile src/filesystem/Makefile doc/Makefile po/Makefile testsuite/Makefile python/Makefile"
+ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/c++17/Makefile src/c++20/Makefile src/filesystem/Makefile doc/Makefile po/Makefile python/Makefile"
 
 
 ac_config_commands="$ac_config_commands generate-headers"
@@ -79271,7 +79271,6 @@
     "src/filesystem/Makefile") CONFIG_FILES="$CONFIG_FILES src/filesystem/Makefile" ;;
     "doc/Makefile") CONFIG_FILES="$CONFIG_FILES doc/Makefile" ;;
     "po/Makefile") CONFIG_FILES="$CONFIG_FILES po/Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
     "python/Makefile") CONFIG_FILES="$CONFIG_FILES python/Makefile" ;;
     "generate-headers") CONFIG_COMMANDS="$CONFIG_COMMANDS generate-headers" ;;
 
diff -urBb gcc-11.2.0-orig/libstdc++-v3/configure.ac gcc-11.2.0/libstdc++-v3/configure.ac
--- gcc-11.2.0-orig/libstdc++-v3/configure.ac	2021-07-28 10:55:09.228314400 +0400
+++ gcc-11.2.0/libstdc++-v3/configure.ac	2021-10-12 11:00:59.552952800 +0400
@@ -470,9 +470,6 @@
 # Check if assembler supports rdseed opcode.
 GLIBCXX_CHECK_X86_RDSEED
 
-# This depends on GLIBCXX_ENABLE_SYMVERS and GLIBCXX_IS_NATIVE.
-GLIBCXX_CONFIGURE_TESTSUITE
-
 # For gthread support.  Depends on GLIBCXX_ENABLE_SYMVERS.
 GLIBCXX_CHECK_GTHREADS
 
diff -urBb gcc-11.2.0-orig/libvtv/Makefile.am gcc-11.2.0/libvtv/Makefile.am
--- gcc-11.2.0-orig/libvtv/Makefile.am	2021-07-28 10:55:09.528318600 +0400
+++ gcc-11.2.0/libvtv/Makefile.am	2021-10-12 11:00:59.568917600 +0400
@@ -19,12 +19,6 @@
 ## along with this library; see the file COPYING3.  If not see
 ## <http://www.gnu.org/licenses/>.
 
-if ENABLE_VTABLE_VERIFY
-  SUBDIRS = testsuite
-else
-  SUBDIRS=
-endif
-
 ACLOCAL_AMFLAGS = -I .. -I ../config
 
 # May be used by toolexeclibdir.
diff -urBb gcc-11.2.0-orig/libvtv/Makefile.in gcc-11.2.0/libvtv/Makefile.in
--- gcc-11.2.0-orig/libvtv/Makefile.in	2021-07-28 10:55:09.528318600 +0400
+++ gcc-11.2.0/libvtv/Makefile.in	2021-10-12 11:00:59.572904300 +0400
@@ -264,7 +264,6 @@
 ETAGS = etags
 CTAGS = ctags
 CSCOPE = cscope
-DIST_SUBDIRS = testsuite
 ACLOCAL = @ACLOCAL@
 AMTAR = @AMTAR@
 AM_DEFAULT_VERBOSITY = @AM_DEFAULT_VERBOSITY@
@@ -405,8 +404,6 @@
 top_srcdir = @top_srcdir@
 toplevel_builddir = @toplevel_builddir@
 toplevel_srcdir = @toplevel_srcdir@
-@ENABLE_VTABLE_VERIFY_FALSE@SUBDIRS = 
-@ENABLE_VTABLE_VERIFY_TRUE@SUBDIRS = testsuite
 ACLOCAL_AMFLAGS = -I .. -I ../config
 
 # May be used by toolexeclibdir.
diff -urBb gcc-11.2.0-orig/libvtv/configure gcc-11.2.0/libvtv/configure
--- gcc-11.2.0-orig/libvtv/configure	2021-07-28 10:55:09.528318600 +0400
+++ gcc-11.2.0/libvtv/configure	2021-10-12 11:00:59.560907200 +0400
@@ -15748,11 +15748,6 @@
 ac_config_files="$ac_config_files Makefile"
 
 
-if test "x$VTV_SUPPORTED" = "xyes"; then
-  ac_config_files="$ac_config_files testsuite/Makefile"
-
-fi
-
 case "$target_os" in
   cygwin*|mingw32*)
     vtv_cygmin="yes"
@@ -16947,7 +16942,6 @@
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
-    "testsuite/Makefile") CONFIG_FILES="$CONFIG_FILES testsuite/Makefile" ;;
 
   *) as_fn_error $? "invalid argument: \`$ac_config_target'" "$LINENO" 5;;
   esac
diff -urBb gcc-11.2.0-orig/libvtv/configure.ac gcc-11.2.0/libvtv/configure.ac
--- gcc-11.2.0-orig/libvtv/configure.ac	2021-07-28 10:55:09.528318600 +0400
+++ gcc-11.2.0/libvtv/configure.ac	2021-10-12 11:00:59.566891900 +0400
@@ -174,21 +174,6 @@
 
 AC_CONFIG_FILES([Makefile])
 
-if test "x$VTV_SUPPORTED" = "xyes"; then
-  AC_CONFIG_FILES(AC_FOREACH([DIR], [testsuite], [DIR/Makefile ]), 
-    [cat > vpsed$$ << \_EOF
-s!`test -f '$<' || echo '$(srcdir)/'`!!
-_EOF
-    sed -f vpsed$$ $ac_file > tmp$$
-    mv tmp$$ $ac_file
-    rm vpsed$$
-    echo 'MULTISUBDIR =' >> $ac_file
-    ml_norecursion=yes
-    . ${multi_basedir}/config-ml.in
-    AS_UNSET([ml_norecursion])
-])
-fi
-
 case "$target_os" in
   cygwin*|mingw32*)
     vtv_cygmin="yes"
