diff -urBb gcc-6.1.0-orig/gcc/config.gcc gcc-6.1.0/gcc/config.gcc
--- gcc-6.1.0-orig/gcc/config.gcc	2016-06-12 10:12:39.166400400 +0400
+++ gcc-6.1.0/gcc/config.gcc	2016-06-12 10:22:08.092318800 +0400
@@ -1710,9 +1710,9 @@
 	fi
 	tm_file="${tm_file} i386/mingw32.h"
 	# This makes the logic if mingw's or the w64 feature set has to be used
+	user_headers_inc_next_post="${user_headers_inc_next_post} float.h"
 	case ${target} in
 		*-w64-*)
-			user_headers_inc_next_post="${user_headers_inc_next_post} float.h"
 			user_headers_inc_next_pre="${user_headers_inc_next_pre} stddef.h stdarg.h"
 			tm_file="${tm_file} i386/mingw-w64.h"
 			if test x$enable_targets = xall; then
