diff -urBbN gcc-6.3.0-orig/gcc/config.gcc gcc-6.3.0/gcc/config.gcc
--- gcc-6.3.0-orig/gcc/config.gcc	2016-11-08 01:38:43.000000000 +0400
+++ gcc-6.3.0/gcc/config.gcc	2016-12-23 20:47:11.171931000 +0400
@@ -1756,7 +1756,10 @@
 			tmake_file="${tmake_file} i386/t-mingw-w32"
 			;;
 	esac
+	# Don't ignore values passed in to configure via --native-system-header-dir
+	if test x$native_system_header_dir = x ; then
         native_system_header_dir=/mingw/include
+	fi
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	extra_options="${extra_options} i386/cygming.opt i386/mingw.opt"
 	case ${target} in
