diff -urBb gcc-5.1.0-orig/libiberty/Makefile.in gcc-5.1.0/libiberty/Makefile.in
--- gcc-5.1.0-orig/libiberty/Makefile.in	2014-12-24 19:22:51.000000000 +0300
+++ gcc-5.1.0/libiberty/Makefile.in	2015-06-04 22:33:34.808756000 +0300
@@ -354,7 +354,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
