diff -urBb tk8.6.17-orig/win/Makefile.in tk8.6.17/win/Makefile.in
--- tk8.6.17-orig/win/Makefile.in	2025-07-31 21:34:04.000000000 +0400
+++ tk8.6.17/win/Makefile.in	2026-03-29 12:58:25.458531200 +0400
@@ -499,6 +499,9 @@
 		$(COPY) $$i "$(LIB_INSTALL_DIR)"; \
 	    fi; \
 	    done
+	@echo "Installing pkg-config file to $(LIB_INSTALL_DIR)/pkgconfig/"
+	@$(MKDIR) "$(LIB_INSTALL_DIR)/pkgconfig"
+	@$(INSTALL_DATA) tk.pc "$(LIB_INSTALL_DIR)/pkgconfig/tk.pc"
 
 install-libraries: libraries
 	@for i in "$$($(CYGPATH) $(prefix)/lib)" \
diff -urBb tk8.6.17-orig/win/configure.in tk8.6.17/win/configure.in
--- tk8.6.17-orig/win/configure.in	2025-07-31 21:34:04.000000000 +0400
+++ tk8.6.17/win/configure.in	2026-03-29 12:58:25.461956000 +0400
@@ -339,6 +339,11 @@
 AC_SUBST(RC_DEFINE)
 AC_SUBST(RC_DEFINES)
 AC_SUBST(RES)
+AC_SUBST(XLIBSW)
+AC_SUBST(XFT_LIBS)
+AC_CONFIG_FILES([
+    tk.pc:../unix/tk.pc.in
+])
 
 AC_CONFIG_FILES([Makefile tkConfig.sh wish.exe.manifest])
 AC_OUTPUT
