diff -urBb wget-1.16.1-orig/src/utils.c wget-1.16.1/src/utils.c
--- wget-1.16.1-orig/src/utils.c	2014-11-23 19:49:06.000000000 +0300
+++ wget-1.16.1/src/utils.c	2015-02-14 21:42:29.656738700 +0300
@@ -64,8 +64,10 @@
 #include <sys/stat.h>
 
 /* For TIOCGWINSZ and friends: */
+#ifndef _WIN32
 #include <sys/ioctl.h>
 #include <termios.h>
+#endif
 
 /* Needed for Unix version of run_with_timeout. */
 #include <signal.h>
