diff -urBb wget-1.15-orig/configure.ac wget-1.15/configure.ac
--- wget-1.15-orig/configure.ac	2014-01-04 16:49:47 +0400
+++ wget-1.15/configure.ac	2014-03-22 12:19:45 +0400
@@ -259,14 +259,14 @@
       AC_CHECK_LIB(eay32, EVP_MD_CTX_init)
       if test x"$ac_cv_lib_eay32_EVP_MD_CTX_init" != xno
       then
-        AC_CHECK_LIB(ssl32, SSL_connect, [
+        AC_CHECK_LIB(ssl, SSL_connect, [
           ssl_found=yes
           AC_MSG_NOTICE([Enabling support for SSL via OpenSSL (shared)])
           AC_LIBOBJ([openssl])
-          LIBS="${LIBS} -lssl32"
-          AC_DEFINE([HAVE_LIBSSL32], [1], [Define to 1 if you have the `ssl32' library (-lssl32).])
+          LIBS="${LIBS} -lssl"
+          AC_DEFINE([HAVE_LIBSSL32], [1], [Define to 1 if you have the `ssl' library (-lssl).])
         ],
-        AC_MSG_ERROR([openssl not found: shared lib eay32 found but ssl32 not found]))
+        AC_MSG_ERROR([openssl not found: shared lib eay32 found but ssl not found]))
           
       else
         LIBS+=' -lgdi32'
