diff -urBb ninja-1.10.2-orig/CMakeLists.txt ninja-1.10.2/CMakeLists.txt
--- ninja-1.10.2-orig/CMakeLists.txt	2020-11-28 15:27:23.000000000 +0400
+++ ninja-1.10.2/CMakeLists.txt	2022-01-02 10:24:51.785798300 +0400
@@ -2,6 +2,7 @@
 
 include(CheckIncludeFileCXX)
 include(CheckIPOSupported)
+include(CheckSymbolExists)
 
 project(ninja)
 
@@ -68,7 +69,18 @@
 
 	# Now check availability of the unistd header
 	check_include_file_cxx(unistd.h PLATFORM_HAS_UNISTD_HEADER)
-	set(${RESULT} "${PLATFORM_HAS_UNISTD_HEADER}" PARENT_SCOPE)
+	if (NOT PLATFORM_HAS_UNISTD_HEADER)
+		set(${RESULT} "0" PARENT_SCOPE)
+		return()
+	endif()
+
+	check_symbol_exists(pipe "unistd.h" HAVE_PIPE)
+	check_symbol_exists(fork "unistd.h" HAVE_FORK)
+	if (HAVE_PIPE AND HAVE_FORK)
+		set(${RESULT} "TRUE" PARENT_SCOPE)
+	else()
+		set(${RESULT} "FALSE" PARENT_SCOPE)
+	endif()
 endfunction()
 
 check_platform_supports_browse_mode(platform_supports_ninja_browse)
