diff -urBb ../icu-orig/source/i18n/digitlst.cpp source/i18n/digitlst.cpp
--- ../icu-orig/source/i18n/digitlst.cpp	2017-01-20 04:20:31.000000000 +0400
+++ source/i18n/digitlst.cpp	2017-09-28 13:41:11.909950300 +0400
@@ -489,7 +489,20 @@
 
 #if U_USE_STRTOD_L && U_PLATFORM_USES_ONLY_WIN32_API
 # define locale_t _locale_t
+// Stubs
+#if defined(__MINGW32__)
+inline  locale_t createlocale(int _Category,const char *_Locale) {
+    locale_t loc;
+    return loc;
+}
+
+inline void freelocale(locale_t _Locale) {
+
+}
+#else
 # define freelocale _free_locale
+# define createlocale _create_locale
+#endif
 # define strtod_l _strtod_l
 #endif
 
@@ -514,7 +527,7 @@
     ucln_i18n_registerCleanup(UCLN_I18N_DIGITLIST, digitList_cleanup);
 #if U_USE_STRTOD_L
 # if U_PLATFORM_USES_ONLY_WIN32_API
-    gCLocale = _create_locale(LC_ALL, "C");
+    gCLocale = createlocale(LC_ALL, "C");
 # else
     gCLocale = newlocale(LC_ALL_MASK, "C", (locale_t)0);
 # endif
diff -urBb ../icu-orig/source/io/ufile.cpp source/io/ufile.cpp
--- ../icu-orig/source/io/ufile.cpp	2017-02-20 01:58:30.000000000 +0400
+++ source/io/ufile.cpp	2017-09-28 13:06:59.920056000 +0400
@@ -160,7 +160,7 @@
         // TODO: test this code path, including wperm.
         wchar_t wperm[40] = {};
         size_t  retVal;
-        mbstowcs_s(&retVal, wperm, perm, _TRUNCATE);
+        mbstowcs_s(&retVal, wperm, 40, perm, _TRUNCATE);
         FILE *systemFile = _wfopen((const wchar_t *)filename, wperm);
         if (systemFile) {
             result = finit_owner(systemFile, locale, codepage, TRUE);
diff -urBb ../icu-orig/source/tools/toolutil/pkg_genc.cpp source/tools/toolutil/pkg_genc.cpp
--- ../icu-orig/source/tools/toolutil/pkg_genc.cpp	2017-04-13 13:46:02.000000000 +0400
+++ source/tools/toolutil/pkg_genc.cpp	2017-09-28 13:10:24.983405000 +0400
@@ -43,6 +43,7 @@
 #include "cmemory.h"
 #include "cstring.h"
 #include "filestrm.h"
+#include "filetools.h"
 #include "toolutil.h"
 #include "unicode/uclean.h"
 #include "uoptions.h"
