diff -urBb glib-2.44.1-orig/config.h.in glib-2.44.1/config.h.in
--- glib-2.44.1-orig/config.h.in	2015-05-13 05:50:38.000000000 +0300
+++ glib-2.44.1/config.h.in	2015-07-15 17:05:01.032409400 +0300
@@ -247,6 +247,9 @@
 /* Define if libelf is available */
 #undef HAVE_LIBELF
 
+/* Define to 1 if you have the `iphlpapi' library (-liphlpapi). */
+#undef HAVE_LIBIPHLPAPI
+
 /* Define to 1 if you have the <linux/magic.h> header file. */
 #undef HAVE_LINUX_MAGIC_H
 
@@ -292,6 +295,9 @@
 /* Define to 1 if you have the <ndir.h> header file, and it defines `DIR'. */
 #undef HAVE_NDIR_H
 
+/* Define to 1 if you have the <netioapi.h> header file. */
+#undef HAVE_NETIOAPI_H
+
 /* We have AF_NETLINK sockets */
 #undef HAVE_NETLINK
 
@@ -590,8 +596,7 @@
 /* Define to 1 if you have the `_NSGetEnviron' function. */
 #undef HAVE__NSGETENVIRON
 
-/* Define to the sub-directory in which libtool stores uninstalled libraries.
-   */
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
 #undef LT_OBJDIR
 
 /* Define to 1 if `major', `minor', and `makedev' are declared in <mkdev.h>.
@@ -605,6 +610,9 @@
 /* didn't find fd_set */
 #undef NO_FD_SET
 
+/* Define to 1 if your C compiler doesn't accept -c and -o together. */
+#undef NO_MINUS_C_MINUS_O
+
 /* Define to the address where bug reports for this package should be sent. */
 #undef PACKAGE_BUGREPORT
 
diff -urBb glib-2.44.1-orig/configure.ac glib-2.44.1/configure.ac
--- glib-2.44.1-orig/configure.ac	2015-05-13 05:45:52.000000000 +0300
+++ glib-2.44.1/configure.ac	2015-07-15 17:03:47.869224700 +0300
@@ -1021,6 +1021,26 @@
 fi
 
 AC_CHECK_FUNCS(getprotobyname_r endservent if_nametoindex if_indextoname sendmmsg)
+if test $glib_native_win32 = yes
+then
+	AC_CHECK_HEADERS(netioapi.h, [], [], [#include <winsock2.h>])
+	if test "x$ac_cv_func_if_nametoindex" != "xyes"
+	then
+	  AC_CHECK_LIB(iphlpapi, if_nametoindex)
+	  if test "x$ac_cv_lib_iphlpapi_if_nametoindex" = "xyes"
+	  then
+		AC_DEFINE(HAVE_IF_NAMETOINDEX)
+	  fi
+	fi
+	if test "x$ac_cv_func_if_indextoname" != "xyes"
+	then
+	  AC_CHECK_LIB(iphlpapi, if_indextoname)
+	  if test "x$ac_cv_lib_iphlpapi_if_indextoname" = "xyes"
+	  then
+		AC_DEFINE(HAVE_IF_INDEXTONAME)
+	  fi
+	fi
+fi
 
 AS_IF([test $glib_native_win32 = yes], [
   # <wspiapi.h> in the Windows SDK and in mingw-w64 has wrappers for
diff -urBb glib-2.44.1-orig/gio/gsocket.c glib-2.44.1/gio/gsocket.c
--- glib-2.44.1-orig/gio/gsocket.c	2015-03-22 21:19:51.000000000 +0300
+++ glib-2.44.1/gio/gsocket.c	2015-07-15 16:43:06.086198800 +0300
@@ -67,6 +67,9 @@
 #ifdef G_OS_WIN32
 /* For Windows XP runtime compatibility, but use the system's if_nametoindex() if available */
 #include "gwin32networking.h"
+#ifdef HAVE_NETIOAPI_H
+#include <netioapi.h>
+#endif
 #endif
 
 /**
