diff -urBb glib-2.44.1-orig/configure.ac glib-2.44.1/configure.ac
--- glib-2.44.1-orig/configure.ac	2015-05-13 05:45:52.000000000 +0300
+++ glib-2.44.1/configure.ac	2015-07-23 16:57:31.337680400 +0300
@@ -1021,6 +1021,40 @@
 fi
 
 AC_CHECK_FUNCS(getprotobyname_r endservent if_nametoindex if_indextoname sendmmsg)
+if test $glib_native_win32 = yes
+then
+  AC_CHECK_HEADERS(netioapi.h, [], [], [#include <winsock2.h>])
+  if test "x$ac_cv_func_if_nametoindex" != "xyes"; then
+    AC_MSG_CHECKING([for if_nametoindex in -liphlpapi])
+    _save_LIBS=${LIBS}
+    LIBS="${LIBS} -liphlpapi"
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#include <winsock2.h>
+#include <netioapi.h>]],
+[[return if_nametoindex("i");]])],
+                     [ac_cv_func_if_nametoindex=yes], [LIBS=${_save_LIBS}])
+    if test x$ac_cv_func_if_nametoindex = xyes; then
+      AC_DEFINE(HAVE_IF_NAMETOINDEX)
+    fi
+    AC_MSG_RESULT($ac_cv_func_if_nametoindex)
+  fi
+  if test "x$ac_cv_func_if_indextoname" != "xyes"; then
+    AC_MSG_CHECKING([for if_indextoname in -liphlpapi])
+    _save_LIBS=${LIBS}
+    if test x$ac_cv_func_if_nametoindex != xyes; then
+      LIBS="${LIBS} -liphlpapi"
+    fi
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#include <winsock2.h>
+#include <netioapi.h>]],
+[[return if_indextoname(0, 0);]])],
+                     [ac_cv_func_if_indextoname=yes], [LIBS=${_save_LIBS}])
+    if test x$ac_cv_func_if_indextoname = xyes; then
+      AC_DEFINE(HAVE_IF_INDEXTONAME)
+    fi
+    AC_MSG_RESULT($ac_cv_func_if_indextoname)
+  fi
+fi
 
 AS_IF([test $glib_native_win32 = yes], [
   # <wspiapi.h> in the Windows SDK and in mingw-w64 has wrappers for
diff -urBb glib-2.44.1-orig/gio/gsocket.c glib-2.44.1/gio/gsocket.c
--- glib-2.44.1-orig/gio/gsocket.c	2015-03-22 21:19:51.000000000 +0300
+++ glib-2.44.1/gio/gsocket.c	2015-07-15 16:43:06.086198800 +0300
@@ -67,6 +67,9 @@
 #ifdef G_OS_WIN32
 /* For Windows XP runtime compatibility, but use the system's if_nametoindex() if available */
 #include "gwin32networking.h"
+#ifdef HAVE_NETIOAPI_H
+#include <netioapi.h>
+#endif
 #endif
 
 /**
