diff -urBb glib-2.58.1-orig/configure.ac glib-2.58.1/configure.ac
--- glib-2.58.1-orig/configure.ac	2018-09-21 18:23:52.000000000 +0400
+++ glib-2.58.1/configure.ac	2018-10-18 21:52:30.708529200 +0400
@@ -961,6 +961,41 @@
           AC_DEFINE(BROKEN_IP_MREQ_SOURCE_STRUCT, 1, [struct ip_mreq_source definition is broken on Android NDK <= r16])
   ])])
 
+if test $glib_native_win32 = yes
+then
+  AC_CHECK_HEADERS(netioapi.h, [], [], [#include <winsock2.h>])
+  if test "x$ac_cv_func_if_nametoindex" != "xyes"; then
+    AC_MSG_CHECKING([for if_nametoindex in -liphlpapi])
+    _save_LIBS=${LIBS}
+    LIBS="${LIBS} -liphlpapi"
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#include <winsock2.h>
+#include <netioapi.h>]],
+[[return if_nametoindex("i");]])],
+                     [ac_cv_func_if_nametoindex=yes], [LIBS=${_save_LIBS}])
+    if test x$ac_cv_func_if_nametoindex = xyes; then
+      AC_DEFINE(HAVE_IF_NAMETOINDEX)
+    fi
+    AC_MSG_RESULT($ac_cv_func_if_nametoindex)
+  fi
+  if test "x$ac_cv_func_if_indextoname" != "xyes"; then
+    AC_MSG_CHECKING([for if_indextoname in -liphlpapi])
+    _save_LIBS=${LIBS}
+    if test x$ac_cv_func_if_nametoindex != xyes; then
+      LIBS="${LIBS} -liphlpapi"
+    fi
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#include <winsock2.h>
+#include <netioapi.h>]],
+[[return if_indextoname(0, 0);]])],
+                     [ac_cv_func_if_indextoname=yes], [LIBS=${_save_LIBS}])
+    if test x$ac_cv_func_if_indextoname = xyes; then
+      AC_DEFINE(HAVE_IF_INDEXTONAME)
+    fi
+    AC_MSG_RESULT($ac_cv_func_if_indextoname)
+  fi
+fi
+
 AS_IF([test $glib_native_win32 = yes], [
   # <wspiapi.h> in the Windows SDK and in mingw-w64 has wrappers for
   # inline workarounds for getaddrinfo, getnameinfo and freeaddrinfo if
diff -urBb glib-2.58.1-orig/gio/gsocket.c glib-2.58.1/gio/gsocket.c
--- glib-2.58.1-orig/gio/gsocket.c	2018-09-21 18:23:52.000000000 +0400
+++ glib-2.58.1/gio/gsocket.c	2018-10-18 21:52:30.716537200 +0400
@@ -78,6 +78,9 @@
 #ifdef G_OS_WIN32
 /* For Windows XP runtime compatibility, but use the system's if_nametoindex() if available */
 #include "gwin32networking.h"
+#ifdef HAVE_NETIOAPI_H
+#include <netioapi.h>
+#endif
 #endif
 
 /**
diff -urBb glib-2.58.1-orig/glib/gstrfuncs.c glib-2.58.1/glib/gstrfuncs.c
--- glib-2.58.1-orig/glib/gstrfuncs.c	2018-09-21 18:23:53.000000000 +0400
+++ glib-2.58.1/glib/gstrfuncs.c	2018-10-18 21:52:30.721541100 +0400
@@ -1282,7 +1282,7 @@
       gchar buf[1024];
       GError *error = NULL;
 
-#if defined(G_OS_WIN32)
+#if defined(G_OS_WIN32) && !defined(__MINGW32__)
       strerror_s (buf, sizeof (buf), errnum);
       msg = buf;
 #elif defined(HAVE_STRERROR_R)
