diff -urBb gcc-11.2.0-orig/libiberty/Makefile.in gcc-11.2.0/libiberty/Makefile.in
--- gcc-11.2.0-orig/libiberty/Makefile.in	2021-07-28 10:55:09.088312400 +0400
+++ gcc-11.2.0/libiberty/Makefile.in	2021-10-12 10:46:42.504798300 +0400
@@ -358,7 +358,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
