diff -urBb gimp-2.6.12-orig/plug-ins/script-fu/script-fu-server.c gimp-2.6.12/plug-ins/script-fu/script-fu-server.c
--- gimp-2.6.12-orig/plug-ins/script-fu/script-fu-server.c	2012-01-31 17:50:28 +0400
+++ gimp-2.6.12/plug-ins/script-fu/script-fu-server.c	2012-02-25 16:00:34 +0400
@@ -36,7 +36,10 @@
 #include <glib.h>
 
 #ifdef G_OS_WIN32
+// for getaddrinfo(), getnameinfo()
+#define _WIN32_WINNT 0x0501
 #include <winsock2.h>
+#include <ws2tcpip.h>
 #include <libgimpbase/gimpwin32-io.h>
 #else
 #include <sys/socket.h>
@@ -131,7 +134,11 @@
 
 typedef union
 {
+#ifdef G_OS_WIN32
+  unsigned short           family;
+#else
   sa_family_t              family;
+#endif
   struct sockaddr_storage  ss;
   struct sockaddr          sa;
   struct sockaddr_in       sa_in;
@@ -331,7 +338,11 @@
       gchar                    clientname[NI_MAXHOST];
 
       /* Connection request on original socket. */
+#ifdef G_OS_WIN32
+      int                    size = sizeof (client);
+#else
       guint                    size = sizeof (client);
+#endif
       gint                     new;
       guint                    portno;
 
@@ -446,7 +457,11 @@
   const gchar     *progress;
 
   memset (&hints, 0, sizeof (hints));
+#ifndef AI_ADDRCONFIG
+  hints.ai_flags = AI_PASSIVE;
+#else
   hints.ai_flags = AI_PASSIVE | AI_ADDRCONFIG;
+#endif
   hints.ai_socktype = SOCK_STREAM;
 
   port_s = g_strdup_printf ("%d", port);
@@ -523,7 +538,11 @@
 static gboolean
 execute_command (SFCommand *cmd)
 {
+#ifdef G_OS_WIN32
+  char    buffer[RESPONSE_HEADER];
+#else
   guchar    buffer[RESPONSE_HEADER];
+#endif
   GString  *response;
   time_t    clock1;
   time_t    clock2;
@@ -584,7 +603,11 @@
 read_from_client (gint filedes)
 {
   SFCommand *cmd;
+#ifdef G_OS_WIN32
+  char     buffer[COMMAND_HEADER];
+#else
   guchar     buffer[COMMAND_HEADER];
+#endif
   gchar     *command;
   gchar     *clientaddr;
   time_t     clock;
@@ -698,7 +721,11 @@
       gimp_quit ();
     }
 
+#ifdef G_OS_WIN32
+  setsockopt (sock, SOL_SOCKET, SO_REUSEADDR, (const char*)&v, sizeof(v));
+#else
   setsockopt (sock, SOL_SOCKET, SO_REUSEADDR, &v, sizeof(v));
+#endif
 
 #ifdef IPV6_V6ONLY
   /* Only listen on IPv6 addresses, otherwise bind() will fail. */
