diff -urBb Python-3.14.3-orig/configure.ac Python-3.14.3/configure.ac
--- Python-3.14.3-orig/configure.ac	2026-03-28 11:38:34.640970000 +0400
+++ Python-3.14.3/configure.ac	2026-03-28 11:40:59.113031000 +0400
@@ -4420,7 +4420,7 @@
 AC_MSG_RESULT([$with_system_libmpdec])
 
 AC_DEFUN([USE_BUNDLED_LIBMPDEC],
-         [LIBMPDEC_CFLAGS="-I\$(srcdir)/Modules/_decimal/libmpdec"
+         [LIBMPDEC_CFLAGS="-idirafter \$(srcdir)/Modules/_decimal/libmpdec"
           LIBMPDEC_LIBS="-lm \$(LIBMPDEC_A)"
           LIBMPDEC_INTERNAL="\$(LIBMPDEC_HEADERS) \$(LIBMPDEC_A)"
           have_mpdec=yes
