diff -urBb binutils-2.20-orig/bfd/Makefile.in binutils-2.20/bfd/Makefile.in
--- binutils-2.20-orig/bfd/Makefile.in	2009-10-16 14:47:48 +0300
+++ binutils-2.20/bfd/Makefile.in	2010-01-03 23:48:47 +0300
@@ -320,7 +320,7 @@
 RELEASE = y
 INCDIR = $(srcdir)/../include
 CSEARCH = -I. -I$(srcdir) -I$(INCDIR)
-SUBDIRS = doc po
+SUBDIRS = po
 bfddocdir = doc
 libbfd_la_LDFLAGS = $(am__append_1) -release `cat libtool-soversion` \
 	@SHARED_LDFLAGS@ $(am__empty)
