diff -urBb postgresql-8.4.3-orig/src/include/port.h postgresql-8.4.3/src/include/port.h
--- postgresql-8.4.3-orig/src/include/port.h	2009-11-14 18:39:45 +0300
+++ postgresql-8.4.3/src/include/port.h	2010-03-31 21:15:06 +0400
@@ -293,6 +293,7 @@
  */
 #if defined(WIN32) && !defined(__CYGWIN__) && !defined(UNSAFE_STAT_OK)
 #include <sys/stat.h>
+#undef stat
 extern int	pgwin32_safestat(const char *path, struct stat * buf);
 
 #define stat(a,b) pgwin32_safestat(a,b)
