diff -urBb gettext-0.18.1.1-orig/gettext-tools/woe32dll/export.h gettext-0.18.1.1/gettext-tools/woe32dll/export.h
--- gettext-0.18.1.1-orig/gettext-tools/woe32dll/export.h	2010-06-06 16:49:58 +0400
+++ gettext-0.18.1.1/gettext-tools/woe32dll/export.h	2012-11-27 16:19:13 +0400
@@ -90,7 +90,11 @@
    --export-all-symbols is used.  */
 
  /* IMP(x) is a symbol that contains the address of x.  */
+#ifdef _WIN64
+#define IMP(x) __imp_##x
+#else
 #define IMP(x) _imp__##x
+#endif
 
  /* Ensure that the variable x is exported from the library, and that a
     pseudo-variable IMP(x) is available.  */
