diff -urBb qtmultimedia-everywhere-src-5.11.2-orig/src/plugins/common/evr/evrdefs.h qtmultimedia-everywhere-src-5.11.2/src/plugins/common/evr/evrdefs.h
--- qtmultimedia-everywhere-src-5.11.2-orig/src/plugins/common/evr/evrdefs.h	2018-08-29 14:16:00.000000000 +0400
+++ qtmultimedia-everywhere-src-5.11.2/src/plugins/common/evr/evrdefs.h	2018-10-21 21:20:39.675743100 +0400
@@ -84,6 +84,7 @@
 #endif
 
 #ifdef __GNUC__
+#ifndef _MFVideoNormalizedRect_
 typedef struct MFVideoNormalizedRect {
     float left;
     float top;
@@ -91,6 +92,7 @@
     float bottom;
 } MFVideoNormalizedRect;
 #endif
+#endif
 
 #include <initguid.h>
 
diff -urBb qtmultimedia-everywhere-src-5.11.2-orig/src/plugins/directshow/directshow.pro qtmultimedia-everywhere-src-5.11.2/src/plugins/directshow/directshow.pro
--- qtmultimedia-everywhere-src-5.11.2-orig/src/plugins/directshow/directshow.pro	2018-08-29 14:16:00.000000000 +0400
+++ qtmultimedia-everywhere-src-5.11.2/src/plugins/directshow/directshow.pro	2018-10-21 21:24:58.446283100 +0400
@@ -16,6 +16,11 @@
     DEFINES += NO_DSHOW_STRSAFE
 }
 
+# Resolve linker error: undefined reference to `IID_IMFVideoPresenter', 'IID_IMFVideoDeviceID'.
+mingw {
+    LIBS_PRIVATE += -lamstrmid
+}
+
 include(common/common.pri)
 qtConfig(directshow-player): include(player/player.pri)
 include(camera/camera.pri)
