diff -urBb qtmultimedia-everywhere-src-5.15.7-orig/src/plugins/directshow/directshow.pro qtmultimedia-everywhere-src-5.15.7/src/plugins/directshow/directshow.pro
--- qtmultimedia-everywhere-src-5.15.7-orig/src/plugins/directshow/directshow.pro	2022-09-12 19:29:55.000000000 +0400
+++ qtmultimedia-everywhere-src-5.15.7/src/plugins/directshow/directshow.pro	2023-01-03 14:24:08.611359800 +0400
@@ -20,6 +20,11 @@
     LIBS_PRIVATE += -lamstrmid
 }
 
+# Resolve linker error: undefined reference to `IID_IMFVideoPresenter', 'IID_IMFVideoDeviceID', 'IID_IMFTopologyServiceLookupClient'.
+mingw {
+    LIBS_PRIVATE += -lamstrmid
+}
+
 include(common/common.pri)
 include(player/player.pri)
 include(camera/camera.pri)
