diff -urBb gcc-6.2.0-orig/libgomp/configure gcc-6.2.0/libgomp/configure
--- gcc-6.2.0-orig/libgomp/configure	2016-08-22 12:04:29.000000000 +0400
+++ gcc-6.2.0/libgomp/configure	2016-08-28 11:32:17.171805200 +0400
@@ -16749,9 +16749,9 @@
 
 ac_config_files="$ac_config_files omp.h omp_lib.h omp_lib.f90 libgomp_f.h"
 
-ac_config_files="$ac_config_files Makefile testsuite/Makefile libgomp.spec"
+ac_config_files="$ac_config_files Makefile libgomp.spec"
 
-ac_config_files="$ac_config_files testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in"
+ac_config_files="$ac_config_files"
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
diff -urBb gcc-6.2.0-orig/libgomp/configure.ac gcc-6.2.0/libgomp/configure.ac
--- gcc-6.2.0-orig/libgomp/configure.ac	2016-08-22 12:04:29.000000000 +0400
+++ gcc-6.2.0/libgomp/configure.ac	2016-08-28 11:32:17.176805300 +0400
@@ -367,6 +367,5 @@
 CFLAGS="$save_CFLAGS"
 
 AC_CONFIG_FILES(omp.h omp_lib.h omp_lib.f90 libgomp_f.h)
-AC_CONFIG_FILES(Makefile testsuite/Makefile libgomp.spec)
-AC_CONFIG_FILES([testsuite/libgomp-test-support.pt.exp:testsuite/libgomp-test-support.exp.in])
+AC_CONFIG_FILES(Makefile libgomp.spec)
 AC_OUTPUT
diff -urBb gcc-6.2.0-orig/libgomp/Makefile.am gcc-6.2.0/libgomp/Makefile.am
--- gcc-6.2.0-orig/libgomp/Makefile.am	2016-08-22 12:04:29.000000000 +0400
+++ gcc-6.2.0/libgomp/Makefile.am	2016-08-28 11:32:17.178055300 +0400
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 
 ## May be used by toolexeclibdir.
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
diff -urBb gcc-6.2.0-orig/libgomp/Makefile.in gcc-6.2.0/libgomp/Makefile.in
--- gcc-6.2.0-orig/libgomp/Makefile.in	2016-08-22 12:04:29.000000000 +0400
+++ gcc-6.2.0/libgomp/Makefile.in	2016-08-28 11:32:17.180555300 +0400
@@ -401,7 +401,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I .. -I ../config
-SUBDIRS = testsuite
+SUBDIRS = 
 gcc_version := $(shell cat $(top_srcdir)/../gcc/BASE-VER)
 search_path = $(addprefix $(top_srcdir)/config/, $(config_path)) $(top_srcdir) \
 	      $(top_srcdir)/../include
diff -urBb gcc-6.2.0-orig/libiberty/configure gcc-6.2.0/libiberty/configure
--- gcc-6.2.0-orig/libiberty/configure	2015-11-09 08:32:08.000000000 +0400
+++ gcc-6.2.0/libiberty/configure	2016-08-28 11:32:17.183055400 +0400
@@ -7369,7 +7369,7 @@
 
 
 # We need multilib support, but only if configuring for the target.
-ac_config_files="$ac_config_files Makefile testsuite/Makefile"
+ac_config_files="$ac_config_files Makefile"
 
 ac_config_commands="$ac_config_commands default"
 
diff -urBb gcc-6.2.0-orig/libiberty/configure.ac gcc-6.2.0/libiberty/configure.ac
--- gcc-6.2.0-orig/libiberty/configure.ac	2015-11-09 08:32:08.000000000 +0400
+++ gcc-6.2.0/libiberty/configure.ac	2016-08-28 11:32:17.186805400 +0400
@@ -727,7 +727,7 @@
 AC_SUBST(htmldir)
 
 # We need multilib support, but only if configuring for the target.
-AC_CONFIG_FILES([Makefile testsuite/Makefile])
+AC_CONFIG_FILES([Makefile])
 AC_CONFIG_COMMANDS([default],
   [[test -z "$CONFIG_HEADERS" || echo timestamp > stamp-h
 if test -n "$CONFIG_FILES"; then
diff -urBb gcc-6.2.0-orig/libiberty/Makefile.in gcc-6.2.0/libiberty/Makefile.in
--- gcc-6.2.0-orig/libiberty/Makefile.in	2016-08-28 11:31:55.160432600 +0400
+++ gcc-6.2.0/libiberty/Makefile.in	2016-08-28 11:32:17.188055500 +0400
@@ -99,7 +99,7 @@
 	"tooldir=$(tooldir)"
 
 # Subdirectories to recurse into. We need to override this during cleaning
-SUBDIRS = testsuite
+SUBDIRS = 
 
 # FIXME: add @BUILD_INFO@ once we're sure it works for everyone.
 all: stamp-picdir stamp-noasandir $(TARGETLIB) required-list all-subdir
diff -urBb gcc-6.2.0-orig/libstdc++-v3/configure gcc-6.2.0/libstdc++-v3/configure
--- gcc-6.2.0-orig/libstdc++-v3/configure	2016-04-05 23:03:46.000000000 +0400
+++ gcc-6.2.0/libstdc++-v3/configure	2016-08-28 11:32:17.208055800 +0400
@@ -4954,7 +4954,7 @@
   # expandable list at autoconf time; the second provides an expandable list
   # (i.e., shell variable) at configure time.
 
-  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/filesystem doc po testsuite python'
+  SUBDIRS='include libsupc++ src src/c++98 src/c++11 src/filesystem doc po python'
 
   # These need to be absolute paths, yet at the same time need to
   # canonicalize only relative paths, because then amd will not unmount
@@ -81375,7 +81375,7 @@
 # append it here.  Only modify Makefiles that have just been created.
 #
 # Also, get rid of this simulated-VPATH thing that automake does.
-ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/filesystem/Makefile doc/Makefile po/Makefile testsuite/Makefile python/Makefile"
+ac_config_files="$ac_config_files include/Makefile libsupc++/Makefile src/Makefile src/c++98/Makefile src/c++11/Makefile src/filesystem/Makefile doc/Makefile po/Makefile python/Makefile"
 
 
 ac_config_commands="$ac_config_commands generate-headers"
diff -urBb gcc-6.2.0-orig/libstdc++-v3/Makefile.am gcc-6.2.0/libstdc++-v3/Makefile.am
--- gcc-6.2.0-orig/libstdc++-v3/Makefile.am	2016-01-04 18:30:50.000000000 +0400
+++ gcc-6.2.0/libstdc++-v3/Makefile.am	2016-08-28 11:32:17.223056000 +0400
@@ -24,7 +24,7 @@
 
 if GLIBCXX_HOSTED
 ## Note that python must come after src.
-  hosted_source = src doc po testsuite python
+  hosted_source = src doc po python
 endif
 
 ## Keep this list sync'd with acinclude.m4:GLIBCXX_CONFIGURE.
diff -urBb gcc-6.2.0-orig/libstdc++-v3/Makefile.in gcc-6.2.0/libstdc++-v3/Makefile.in
--- gcc-6.2.0-orig/libstdc++-v3/Makefile.in	2015-07-24 11:05:19.000000000 +0400
+++ gcc-6.2.0/libstdc++-v3/Makefile.in	2016-08-28 11:32:17.224306000 +0400
@@ -55,7 +55,6 @@
 	$(srcdir)/Makefile.in $(srcdir)/Makefile.am \
 	$(top_srcdir)/configure $(am__configure_deps) \
 	$(srcdir)/config.h.in $(srcdir)/../mkinstalldirs \
-	$(top_srcdir)/scripts/testsuite_flags.in \
 	$(top_srcdir)/scripts/extract_symvers.in \
 	$(top_srcdir)/doc/xsl/customization.xsl.in
 subdir = .
@@ -116,7 +115,7 @@
 	$(RECURSIVE_CLEAN_TARGETS:-recursive=) tags TAGS ctags CTAGS
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = include libsupc++ src doc po testsuite python
+DIST_SUBDIRS = include libsupc++ src doc po python
 ABI_TWEAKS_SRCDIR = @ABI_TWEAKS_SRCDIR@
 ACLOCAL = @ACLOCAL@
 ALLOCATOR_H = @ALLOCATOR_H@
@@ -339,7 +338,7 @@
 
 # -I/-D flags to pass when compiling.
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES) $(CPPFLAGS)
-@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po testsuite python
+@GLIBCXX_HOSTED_TRUE@hosted_source = src doc po python
 SUBDIRS = include libsupc++ $(hosted_source)
 ACLOCAL_AMFLAGS = -I . -I .. -I ../config
 @BUILD_XML_FALSE@STAMP_XML = 
