diff -urBb gcc-4.9.2-orig/libiberty/Makefile.in gcc-4.9.2/libiberty/Makefile.in
--- gcc-4.9.2-orig/libiberty/Makefile.in	2013-06-01 03:20:49 +0300
+++ gcc-4.9.2/libiberty/Makefile.in	2015-01-10 00:49:52 +0300
@@ -344,7 +344,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
