diff -urBb qt-creator-opensource-src-3.6.0-orig/src/libs/sqlite/sqlite.pro qt-creator-opensource-src-3.6.0/src/libs/sqlite/sqlite.pro
--- qt-creator-opensource-src-3.6.0-orig/src/libs/sqlite/sqlite.pro	2015-12-11 15:45:01.000000000 +0300
+++ qt-creator-opensource-src-3.6.0/src/libs/sqlite/sqlite.pro	2016-01-06 00:37:23.848267200 +0300
@@ -3,7 +3,13 @@
 
 include(../../qtcreatorlibrary.pri)
 
-win32:DEFINES += SQLITE_API=__declspec(dllexport)
+win32 {
+  equals(MINGW_IN_SHELL, 1) {
+    DEFINES += SQLITE_API=\"__declspec(dllexport)\"
+  } else {
+    DEFINES += SQLITE_API=__declspec(dllexport)
+  }
+}
 unix:DEFINES += SQLITE_API=\"__attribute__((visibility(\\\"default\\\")))\"
 
 include(sqlite-lib.pri)
diff -urBb qt-creator-opensource-src-3.6.0-orig/src/libs/utils/tcpportsgatherer.cpp qt-creator-opensource-src-3.6.0/src/libs/utils/tcpportsgatherer.cpp
--- qt-creator-opensource-src-3.6.0-orig/src/libs/utils/tcpportsgatherer.cpp	2015-12-11 15:45:01.000000000 +0300
+++ qt-creator-opensource-src-3.6.0/src/libs/utils/tcpportsgatherer.cpp	2016-01-06 00:25:21.360920000 +0300
@@ -47,7 +47,8 @@
 #if defined(Q_OS_WIN) && defined(Q_CC_MINGW)
 
 // Missing declarations for MinGW 32.
-#if __GNUC__ == 4 && (!defined(__MINGW64_VERSION_MAJOR) || __MINGW64_VERSION_MAJOR < 2)
+// In mingw-w64-4.0.2 `MIB_TCP_STATE' definition moved to tcpmib.h under '#if (_WIN32_WINNT >= 0x0600)' directive.
+#if (__GNUC__ == 4 || __GNUC__ == 5) && (!defined(__MINGW64_VERSION_MAJOR) || __MINGW64_VERSION_MAJOR < 2 || __MINGW64_VERSION_MAJOR == 4)
 typedef enum { } MIB_TCP_STATE;
 #endif
 
diff -urBb qt-creator-opensource-src-3.6.0-orig/src/tools/winrtdebughelper/winrtdebughelper.cpp qt-creator-opensource-src-3.6.0/src/tools/winrtdebughelper/winrtdebughelper.cpp
--- qt-creator-opensource-src-3.6.0-orig/src/tools/winrtdebughelper/winrtdebughelper.cpp	2015-12-11 15:45:01.000000000 +0300
+++ qt-creator-opensource-src-3.6.0/src/tools/winrtdebughelper/winrtdebughelper.cpp	2016-01-06 00:25:21.363420100 +0300
@@ -71,7 +71,12 @@
 
     const size_t msgBufferSize = 15;
     char pidMessageBuffer[msgBufferSize];
-    int length = sprintf_s(pidMessageBuffer, msgBufferSize, "PID:%d", pid);
+    int length = 
+#ifdef _MSC_VER
+		sprintf_s(pidMessageBuffer, msgBufferSize, "PID:%d", pid);
+#else
+		snprintf(pidMessageBuffer, msgBufferSize, "PID:%d", pid);
+#endif
     if (length >= 0)
         WriteFile(pipe, pidMessageBuffer, length, NULL, NULL);
 
