diff -urBb qt-creator-opensource-src-8.0.2-orig/src/plugins/clangformat/clangformatutils.cpp qt-creator-opensource-src-8.0.2/src/plugins/clangformat/clangformatutils.cpp
--- qt-creator-opensource-src-8.0.2-orig/src/plugins/clangformat/clangformatutils.cpp	2022-10-19 17:33:27.000000000 +0400
+++ qt-creator-opensource-src-8.0.2/src/plugins/clangformat/clangformatutils.cpp	2023-01-03 23:24:28.052135400 +0400
@@ -118,7 +118,10 @@
     style.ColumnLimit = 100;
     style.CommentPragmas = "^ IWYU pragma:";
     style.CompactNamespaces = false;
+#if LLVM_VERSION_MAJOR < 15
+    // deprecated in 15.0
     style.ConstructorInitializerAllOnOneLineOrOnePerLine = false;
+#endif
     style.ConstructorInitializerIndentWidth = 4;
     style.ContinuationIndentWidth = 4;
     style.Cpp11BracedListStyle = true;
