diff -urBb qt-creator-opensource-src-4.6.1-orig/src/libs/sqlite/sqlite.pro qt-creator-opensource-src-4.6.1/src/libs/sqlite/sqlite.pro
--- qt-creator-opensource-src-4.6.1-orig/src/libs/sqlite/sqlite.pro	2018-04-30 16:10:55.000000000 +0400
+++ qt-creator-opensource-src-4.6.1/src/libs/sqlite/sqlite.pro	2018-05-24 23:21:01.661630500 +0400
@@ -3,7 +3,13 @@
 
 include(../../qtcreatorlibrary.pri)
 
-win32:DEFINES += SQLITE_API=__declspec(dllexport)
+win32 {
+  equals(MINGW_IN_SHELL, 1) {
+    DEFINES += SQLITE_API=\"__declspec(dllexport)\"
+  } else {
+    DEFINES += SQLITE_API=__declspec(dllexport)
+  }
+}
 unix:DEFINES += SQLITE_API=\"__attribute__((visibility(\\\"default\\\")))\"
 
 include(sqlite-lib.pri)
diff -urBb qt-creator-opensource-src-4.6.1-orig/src/tools/winrtdebughelper/winrtdebughelper.cpp qt-creator-opensource-src-4.6.1/src/tools/winrtdebughelper/winrtdebughelper.cpp
--- qt-creator-opensource-src-4.6.1-orig/src/tools/winrtdebughelper/winrtdebughelper.cpp	2018-04-30 16:10:55.000000000 +0400
+++ qt-creator-opensource-src-4.6.1/src/tools/winrtdebughelper/winrtdebughelper.cpp	2018-05-24 23:21:01.663583200 +0400
@@ -66,7 +66,12 @@
 
     const size_t msgBufferSize = 15;
     char pidMessageBuffer[msgBufferSize];
-    int length = sprintf_s(pidMessageBuffer, msgBufferSize, "PID:%d", pid);
+    int length = 
+#ifdef _MSC_VER
+		sprintf_s(pidMessageBuffer, msgBufferSize, "PID:%d", pid);
+#else
+		snprintf(pidMessageBuffer, msgBufferSize, "PID:%d", pid);
+#endif
     if (length >= 0)
         WriteFile(pipe, pidMessageBuffer, length, NULL, NULL);
 
