diff -urBb mingw-w64-headers-orig/include/psdk_inc/intrin-impl.h mingw-w64-headers/include/psdk_inc/intrin-impl.h
--- mingw-w64-headers-orig/include/psdk_inc/intrin-impl.h	2017-10-27 14:04:58.000000000 +0400
+++ mingw-w64-headers/include/psdk_inc/intrin-impl.h	2018-05-21 12:24:41.980376400 +0400
@@ -1406,6 +1406,7 @@
 #endif /* __INTRINSIC_PROLOG */
 
 /* NOTE: This should be in immintrin.h */
+#if defined(__GNUC__) && __GNUC__ < 8	// Since gcc-8 _xgetbv in xsaveintrin.h
 #if __INTRINSIC_PROLOG(_xgetbv)
 unsigned __int64 _xgetbv(unsigned int);
 __INTRINSICS_USEINLINE
@@ -1426,6 +1427,7 @@
 }
 #define __INTRINSIC_DEFINED__xgetbv
 #endif /* __INTRINSIC_PROLOG */
+#endif /* defined(__GNUC__) && __GNUC__ < 8 */
 
 #endif /* defined(__x86_64__) || defined(_AMD64_) || defined(__i386__) || defined(_X86_) */
 
