diff -urBb libjpeg-turbo-2.1.4-orig/jconfig.h.in libjpeg-turbo-2.1.4/jconfig.h.in
--- libjpeg-turbo-2.1.4-orig/jconfig.h.in	2022-08-09 01:03:55.000000000 +0400
+++ libjpeg-turbo-2.1.4/jconfig.h.in	2023-01-03 02:36:09.280014400 +0400
@@ -35,3 +35,11 @@
 /* Define if your (broken) compiler shifts signed values as if they were
    unsigned. */
 #cmakedefine RIGHT_SHIFT_IS_UNSIGNED 1
+
+/* Define "boolean" as unsigned char, not int, on Windows systems. */
+#ifdef _WIN32
+#ifndef __RPCNDR_H__		/* don't conflict if rpcndr.h already read */
+typedef unsigned char boolean;
+#endif
+#define HAVE_BOOLEAN		/* prevent jmorecfg.h from redefining it */
+#endif
