diff -urBb libjpeg-turbo-2.0.0-orig/jconfig.h.in libjpeg-turbo-2.0.0/jconfig.h.in
--- libjpeg-turbo-2.0.0-orig/jconfig.h.in	2018-07-27 20:47:48.000000000 +0400
+++ libjpeg-turbo-2.0.0/jconfig.h.in	2018-10-19 11:38:06.981650000 +0400
@@ -66,6 +66,14 @@
   #cmakedefine __CHAR_UNSIGNED__
 #endif
 
+/* Define "boolean" as unsigned char, not int, on Windows systems. */
+#ifdef _WIN32
+#ifndef __RPCNDR_H__		/* don't conflict if rpcndr.h already read */
+typedef unsigned char boolean;
+#endif
+#define HAVE_BOOLEAN		/* prevent jmorecfg.h from redefining it */
+#endif
+ 
 /* Define to empty if `const' does not conform to ANSI C. */
 /* #undef const */
 
