diff -urBb libjpeg-turbo-1.4.0-orig/jconfig.h.in libjpeg-turbo-1.4.0/jconfig.h.in
--- libjpeg-turbo-1.4.0-orig/jconfig.h.in	2014-08-10 03:06:07.000000000 +0400
+++ libjpeg-turbo-1.4.0/jconfig.h.in	2015-07-24 11:55:24.864621700 +0300
@@ -63,6 +63,14 @@
 # undef __CHAR_UNSIGNED__
 #endif
 
+/* Define "boolean" as unsigned char, not int, on Windows systems. */
+#ifdef _WIN32
+#ifndef __RPCNDR_H__		/* don't conflict if rpcndr.h already read */
+typedef unsigned char boolean;
+#endif
+#define HAVE_BOOLEAN		/* prevent jmorecfg.h from redefining it */
+#endif
+ 
 /* Define to empty if `const' does not conform to ANSI C. */
 #undef const
 
Только в libjpeg-turbo-1.4.0: jconfig.h.in.orig
diff -urBb libjpeg-turbo-1.4.0-orig/jmorecfg.h libjpeg-turbo-1.4.0/jmorecfg.h
--- libjpeg-turbo-1.4.0-orig/jmorecfg.h	2014-11-25 13:07:43.000000000 +0300
+++ libjpeg-turbo-1.4.0/jmorecfg.h	2015-07-24 12:00:20.241077800 +0300
@@ -140,16 +140,18 @@
 #endif /* HAVE_UNSIGNED_SHORT */
 
 /* INT16 must hold at least the values -32768..32767. */
-
-#ifndef XMD_H                   /* X11/xmd.h correctly defines INT16 */
-typedef short INT16;
-#endif
-
 /* INT32 must hold at least signed 32-bit values. */
 
 #ifndef XMD_H                   /* X11/xmd.h correctly defines INT32 */
+#ifndef _BASETSD_H_             /* Microsoft defines it in basetsd.h */
+#ifndef _BASETSD_H              /* MinGW is slightly different */
+#ifndef QGLOBAL_H               /* Qt defines it in qglobal.h */
+typedef short INT16;
 typedef long INT32;
 #endif
+#endif
+#endif
+#endif
 
 /* Datatype used for image dimensions.  The JPEG standard only supports
  * images up to 64K*64K due to 16-bit fields in SOF markers.  Therefore
