diff -urBb fontconfig-2.10.2-orig/conf.d/Makefile.am fontconfig-2.10.2/conf.d/Makefile.am
--- fontconfig-2.10.2-orig/conf.d/Makefile.am	2012-10-30 04:53:37 +0400
+++ fontconfig-2.10.2/conf.d/Makefile.am	2013-04-13 14:50:47 +0400
@@ -78,6 +78,15 @@
 
 install-data-hook:
 	mkdir -p $(DESTDIR)$(configdir)
+if OS_WIN32
+	@(echo cd $(DESTDIR)$(configdir);			\
+	  cd $(DESTDIR)$(configdir);				\
+	  for i in $(CONF_LINKS); do				\
+	    echo $(RM) $$i";" cp -p $(DESTDIR)$(templatedir)/$$i .;	\
+	    $(RM) $$i;								\
+	    cp -p $(DESTDIR)$(templatedir)/$$i .;	\
+	  done)
+else
 	@(echo cd $(DESTDIR)$(configdir);			\
 	  cd $(DESTDIR)$(configdir);				\
 	  for i in $(CONF_LINKS); do				\
@@ -85,6 +94,8 @@
 	    $(RM) $$i;						\
 	    ln -s $(templatedir)/$$i .;				\
 	  done)
+endif
+
 uninstall-local:
 	@(echo cd $(DESTDIR)$(configdir);			\
 	  cd $(DESTDIR)$(configdir);				\
