diff -urBb fontconfig-2.13.1-orig/configure.ac fontconfig-2.13.1/configure.ac
--- fontconfig-2.13.1-orig/configure.ac	2018-08-30 12:20:15.000000000 +0400
+++ fontconfig-2.13.1/configure.ac	2018-10-18 22:32:03.856156200 +0400
@@ -814,7 +814,6 @@
 its/Makefile
 po/Makefile.in
 po-conf/Makefile.in
-test/Makefile
 fontconfig.spec
 fontconfig.pc
 fontconfig-zip
diff -urBb fontconfig-2.13.1-orig/Makefile.am fontconfig-2.13.1/Makefile.am
--- fontconfig-2.13.1-orig/Makefile.am	2018-06-05 14:36:38.000000000 +0400
+++ fontconfig-2.13.1/Makefile.am	2018-10-18 22:31:11.427849700 +0400
@@ -24,7 +24,7 @@
 SUBDIRS=fontconfig fc-case fc-lang src \
 	fc-cache fc-cat fc-conflist fc-list fc-match \
 	fc-pattern fc-query fc-scan fc-validate conf.d \
-	its po po-conf test
+	its po po-conf
 if ENABLE_DOCS
 SUBDIRS += doc
 endif
