diff -urBb gdb-9.1-orig/gdb/configure gdb-9.1/gdb/configure
--- gdb-9.1-orig/gdb/configure	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/gdb/configure	2020-05-04 10:29:22.735546700 +0400
@@ -16360,7 +16360,7 @@
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
-    WIN32LIBS="$WIN32LIBS -lws2_32"
+    WIN32LIBS="$WIN32LIBS -lws2_32 -lssp"
     ;;
 esac
 
diff -urBb gdb-9.1-orig/gdb/configure.ac gdb-9.1/gdb/configure.ac
--- gdb-9.1-orig/gdb/configure.ac	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/gdb/configure.ac	2020-05-04 10:29:22.741561200 +0400
@@ -1834,7 +1834,7 @@
 	       POSIX API.  On Windows, we use the Windows API when
 	       building for MinGW, but the POSIX API when building
 	       for Cygwin.])
-    WIN32LIBS="$WIN32LIBS -lws2_32"
+    WIN32LIBS="$WIN32LIBS -lws2_32 -lssp"
     ;;
 esac
 AC_SUBST(WIN32LIBS)
Только в gdb-9.1/gdb: configure.ac.orig
Только в gdb-9.1/gdb: configure.orig
diff -urBb gdb-9.1-orig/gdb/gdbserver/configure gdb-9.1/gdb/gdbserver/configure
--- gdb-9.1-orig/gdb/gdbserver/configure	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/gdb/gdbserver/configure	2020-05-04 10:31:10.425382700 +0400
@@ -9233,6 +9233,12 @@
   done
 fi
 
+if test "${srv_mingw}" = "yes"; then
+  # fixes for mingw-w64-7.0.0
+  # added implementation of __memcpy_chk() and friends
+  srv_libs="${srv_libs} -lssp"
+fi
+
 GDBSERVER_DEPFILES="$srv_regobj $srv_tgtobj $srv_hostio_err_objs $srv_thread_depfiles $srv_selftest_objs"
 GDBSERVER_LIBS="$srv_libs"
 
diff -urBb gdb-9.1-orig/gdb/gdbserver/configure.ac gdb-9.1/gdb/gdbserver/configure.ac
--- gdb-9.1-orig/gdb/gdbserver/configure.ac	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/gdb/gdbserver/configure.ac	2020-05-04 10:32:22.477092700 +0400
@@ -389,6 +389,12 @@
   done
 fi
 
+if test "${srv_mingw}" = "yes"; then
+  # fixes for mingw-w64-7.0.0
+  # added implementation of __memcpy_chk() and friends
+  srv_libs="${srv_libs} -lssp"
+fi
+
 GDBSERVER_DEPFILES="$srv_regobj $srv_tgtobj $srv_hostio_err_objs $srv_thread_depfiles $srv_selftest_objs"
 GDBSERVER_LIBS="$srv_libs"
 
