diff -urBb gdb-9.1-orig/bfd/Makefile.in gdb-9.1/bfd/Makefile.in
--- gdb-9.1-orig/bfd/Makefile.in	2020-02-08 16:50:13.000000000 +0400
+++ gdb-9.1/bfd/Makefile.in	2020-05-04 10:51:36.022152700 +0400
@@ -1761,11 +1761,10 @@
 	for dir in "$(DESTDIR)$(bfdlibdir)" "$(DESTDIR)$(bfdincludedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
-install: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) install-recursive
-install-exec: install-exec-recursive
-install-data: install-data-recursive
-uninstall: uninstall-recursive
+install:
+install-exec:
+install-data:
+uninstall:
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
diff -urBb gdb-9.1-orig/libctf/Makefile.in gdb-9.1/libctf/Makefile.in
--- gdb-9.1-orig/libctf/Makefile.in	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/libctf/Makefile.in	2020-05-04 10:52:21.850623700 +0400
@@ -848,10 +848,10 @@
 	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
-install: install-am
-install-exec: install-exec-am
-install-data: install-data-am
-uninstall: uninstall-am
+install:
+install-exec:
+install-data:
+uninstall:
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
diff -urBb gdb-9.1-orig/libiberty/Makefile.in gdb-9.1/libiberty/Makefile.in
--- gdb-9.1-orig/libiberty/Makefile.in	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/libiberty/Makefile.in	2020-05-04 10:51:36.024147100 +0400
@@ -355,7 +355,7 @@
 @MAINT@	echo stamp > stamp-functions
 
 INSTALL_DEST = @INSTALL_DEST@
-install: install_to_$(INSTALL_DEST) install-subdir
+install: install-subdir
 install-strip: install
 
 .PHONY: install install-strip
diff -urBb gdb-9.1-orig/opcodes/Makefile.in gdb-9.1/opcodes/Makefile.in
--- gdb-9.1-orig/opcodes/Makefile.in	2020-02-08 16:50:14.000000000 +0400
+++ gdb-9.1/opcodes/Makefile.in	2020-05-04 10:51:36.025144300 +0400
@@ -1243,10 +1243,10 @@
 	for dir in "$(DESTDIR)$(bfdlibdir)" "$(DESTDIR)$(bfdincludedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
-install: install-recursive
-install-exec: install-exec-recursive
-install-data: install-data-recursive
-uninstall: uninstall-recursive
+install: 
+install-exec: 
+install-data: 
+uninstall: 
 
 install-am: all-am
 	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
