diff -urBb gdb-8.3.1-orig/gdb/configure gdb-8.3.1/gdb/configure
--- gdb-8.3.1-orig/gdb/configure	2019-09-21 02:04:41.000000000 +0400
+++ gdb-8.3.1/gdb/configure	2020-01-19 16:23:41.982897900 +0400
@@ -15781,7 +15781,7 @@
 
 $as_echo "#define USE_WIN32API 1" >>confdefs.h
 
-    WIN32LIBS="$WIN32LIBS -lws2_32"
+    WIN32LIBS="$WIN32LIBS -lws2_32 -lssp"
     ;;
 esac
 
diff -urBb gdb-8.3.1-orig/gdb/configure.ac gdb-8.3.1/gdb/configure.ac
--- gdb-8.3.1-orig/gdb/configure.ac	2019-09-21 02:04:41.000000000 +0400
+++ gdb-8.3.1/gdb/configure.ac	2020-01-19 16:20:15.115200900 +0400
@@ -1926,7 +1926,7 @@
 	       POSIX API.  On Windows, we use the Windows API when
 	       building for MinGW, but the POSIX API when building
 	       for Cygwin.])
-    WIN32LIBS="$WIN32LIBS -lws2_32"
+    WIN32LIBS="$WIN32LIBS -lws2_32 -lssp"
     ;;
 esac
 AC_SUBST(WIN32LIBS)
diff -urBb gdb-8.3.1-orig/gdb/gdbserver/configure gdb-8.3.1/gdb/gdbserver/configure
--- gdb-8.3.1-orig/gdb/gdbserver/configure	2019-09-21 02:04:41.000000000 +0400
+++ gdb-8.3.1/gdb/gdbserver/configure	2020-01-25 10:28:58.075897800 +0400
@@ -8424,6 +8424,12 @@
   done
 fi
 
+if test "${srv_mingw}" = "yes"; then
+  # fixes for mingw-w64-7.0.0
+  # added implementation of __memcpy_chk() and friends
+  srv_libs="${srv_libs} -lssp"
+fi
+
 GDBSERVER_DEPFILES="$srv_regobj $srv_tgtobj $srv_hostio_err_objs $srv_thread_depfiles $srv_host_obs $srv_selftest_objs"
 GDBSERVER_LIBS="$srv_libs"
 
diff -urBb gdb-8.3.1-orig/gdb/gdbserver/configure.ac gdb-8.3.1/gdb/gdbserver/configure.ac
--- gdb-8.3.1-orig/gdb/gdbserver/configure.ac	2019-09-21 02:04:41.000000000 +0400
+++ gdb-8.3.1/gdb/gdbserver/configure.ac	2020-01-25 10:20:11.228265100 +0400
@@ -402,6 +402,12 @@
   done
 fi
 
+if test "${srv_mingw}" = "yes"; then
+  # fixes for mingw-w64-7.0.0
+  # added implementation of __memcpy_chk() and friends
+  srv_libs="${srv_libs} -lssp"
+fi
+
 GDBSERVER_DEPFILES="$srv_regobj $srv_tgtobj $srv_hostio_err_objs $srv_thread_depfiles $srv_host_obs $srv_selftest_objs"
 GDBSERVER_LIBS="$srv_libs"
 
